{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0807e37b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from sklearn.experimental import enable_iterative_imputer\n",
    "from sklearn.impute import IterativeImputer\n",
    "from sklearn.ensemble import RandomForestRegressor\n",
    "from scipy.stats import gmean, ttest_ind\n",
    "from statsmodels.stats.multitest import multipletests\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from scipy.cluster.hierarchy import dendrogram, linkage\n",
    "from scipy.stats import f_oneway\n",
    "import scipy.stats as ss\n",
    "from statsmodels.stats.multicomp import pairwise_tukeyhsd\n",
    "import statsmodels.api as sm\n",
    "\n",
    "\n",
    "# transcriptome\n",
    "from statsmodels.formula.api import ols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "eb3b45e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "proteome = pd.read_excel(\"/Users/mortezaabyadeh/Desktop/Proteomic.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ede04876",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Master</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Description</th>\n",
       "      <th>Coverage [%]</th>\n",
       "      <th>Abundance Ratio: (Etoposide) / (Control)</th>\n",
       "      <th>Abundance Ratio: (FTY720) / (Control)</th>\n",
       "      <th>Abundance Ratio: (FTY720_plus_Etoposide) / (Control)</th>\n",
       "      <th># PSMs</th>\n",
       "      <th># Peptides</th>\n",
       "      <th># Unique Peptides</th>\n",
       "      <th>...</th>\n",
       "      <th>Abundances (Normalized): F4: Sample, Etoposide</th>\n",
       "      <th>Abundances (Normalized): F5: Sample, Etoposide</th>\n",
       "      <th>Abundances (Normalized): F6: Sample, Etoposide</th>\n",
       "      <th>Abundances (Normalized): F10: Sample, FTY720</th>\n",
       "      <th>Abundances (Normalized): F11: Sample, FTY720</th>\n",
       "      <th>Abundances (Normalized): F12: Sample, FTY720</th>\n",
       "      <th>Abundances (Normalized): F7: Sample, FTY720_plus_Etoposide</th>\n",
       "      <th>Abundances (Normalized): F8: Sample, FTY720_plus_Etoposide</th>\n",
       "      <th>Abundances (Normalized): F9: Sample, FTY720_plus_Etoposide</th>\n",
       "      <th>Contaminant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Master Protein</td>\n",
       "      <td>P08670</td>\n",
       "      <td>Vimentin [OS=Homo sapiens]</td>\n",
       "      <td>90</td>\n",
       "      <td>0.690</td>\n",
       "      <td>1.034</td>\n",
       "      <td>0.956</td>\n",
       "      <td>4919</td>\n",
       "      <td>73</td>\n",
       "      <td>65</td>\n",
       "      <td>...</td>\n",
       "      <td>1.194626e+10</td>\n",
       "      <td>8.138632e+09</td>\n",
       "      <td>8.359638e+09</td>\n",
       "      <td>1.037863e+10</td>\n",
       "      <td>1.260664e+10</td>\n",
       "      <td>1.303859e+10</td>\n",
       "      <td>1.286167e+10</td>\n",
       "      <td>1.279744e+10</td>\n",
       "      <td>1.327241e+10</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Master Protein</td>\n",
       "      <td>Q09666</td>\n",
       "      <td>Neuroblast differentiation-associated protein ...</td>\n",
       "      <td>70</td>\n",
       "      <td>0.833</td>\n",
       "      <td>1.101</td>\n",
       "      <td>1.019</td>\n",
       "      <td>2991</td>\n",
       "      <td>269</td>\n",
       "      <td>268</td>\n",
       "      <td>...</td>\n",
       "      <td>1.772634e+09</td>\n",
       "      <td>2.325231e+09</td>\n",
       "      <td>2.399826e+09</td>\n",
       "      <td>2.122741e+09</td>\n",
       "      <td>2.138561e+09</td>\n",
       "      <td>2.067453e+09</td>\n",
       "      <td>1.941775e+09</td>\n",
       "      <td>2.051846e+09</td>\n",
       "      <td>1.983623e+09</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Master Protein</td>\n",
       "      <td>P60709</td>\n",
       "      <td>Actin, cytoplasmic 1 [OS=Homo sapiens]</td>\n",
       "      <td>94</td>\n",
       "      <td>0.687</td>\n",
       "      <td>0.955</td>\n",
       "      <td>0.888</td>\n",
       "      <td>2652</td>\n",
       "      <td>35</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>7.444967e+09</td>\n",
       "      <td>5.882621e+09</td>\n",
       "      <td>5.694489e+09</td>\n",
       "      <td>5.942383e+09</td>\n",
       "      <td>7.232767e+09</td>\n",
       "      <td>6.858702e+09</td>\n",
       "      <td>6.464013e+09</td>\n",
       "      <td>6.143735e+09</td>\n",
       "      <td>6.091506e+09</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Master Protein</td>\n",
       "      <td>P63261</td>\n",
       "      <td>Actin, cytoplasmic 2 [OS=Homo sapiens]</td>\n",
       "      <td>94</td>\n",
       "      <td>0.414</td>\n",
       "      <td>0.940</td>\n",
       "      <td>0.658</td>\n",
       "      <td>2651</td>\n",
       "      <td>35</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1.120958e+06</td>\n",
       "      <td>1.139723e+06</td>\n",
       "      <td>1.601739e+06</td>\n",
       "      <td>3.598291e+06</td>\n",
       "      <td>1.953038e+06</td>\n",
       "      <td>6.898486e+05</td>\n",
       "      <td>1.758126e+06</td>\n",
       "      <td>1.617831e+06</td>\n",
       "      <td>1.715513e+06</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Master Protein</td>\n",
       "      <td>P04406</td>\n",
       "      <td>Glyceraldehyde-3-phosphate dehydrogenase [OS=H...</td>\n",
       "      <td>76</td>\n",
       "      <td>0.364</td>\n",
       "      <td>0.755</td>\n",
       "      <td>0.755</td>\n",
       "      <td>2418</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>1.199184e+09</td>\n",
       "      <td>8.301472e+08</td>\n",
       "      <td>9.464704e+08</td>\n",
       "      <td>8.816873e+08</td>\n",
       "      <td>9.276548e+08</td>\n",
       "      <td>1.222988e+09</td>\n",
       "      <td>1.461613e+09</td>\n",
       "      <td>1.278716e+09</td>\n",
       "      <td>1.370481e+09</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Master Accession  \\\n",
       "0  Master Protein    P08670   \n",
       "1  Master Protein    Q09666   \n",
       "2  Master Protein    P60709   \n",
       "3  Master Protein    P63261   \n",
       "4  Master Protein    P04406   \n",
       "\n",
       "                                         Description  Coverage [%]  \\\n",
       "0                         Vimentin [OS=Homo sapiens]            90   \n",
       "1  Neuroblast differentiation-associated protein ...            70   \n",
       "2             Actin, cytoplasmic 1 [OS=Homo sapiens]            94   \n",
       "3             Actin, cytoplasmic 2 [OS=Homo sapiens]            94   \n",
       "4  Glyceraldehyde-3-phosphate dehydrogenase [OS=H...            76   \n",
       "\n",
       "   Abundance Ratio: (Etoposide) / (Control)  \\\n",
       "0                                     0.690   \n",
       "1                                     0.833   \n",
       "2                                     0.687   \n",
       "3                                     0.414   \n",
       "4                                     0.364   \n",
       "\n",
       "   Abundance Ratio: (FTY720) / (Control)  \\\n",
       "0                                  1.034   \n",
       "1                                  1.101   \n",
       "2                                  0.955   \n",
       "3                                  0.940   \n",
       "4                                  0.755   \n",
       "\n",
       "   Abundance Ratio: (FTY720_plus_Etoposide) / (Control)  # PSMs  # Peptides  \\\n",
       "0                                              0.956       4919          73   \n",
       "1                                              1.019       2991         269   \n",
       "2                                              0.888       2652          35   \n",
       "3                                              0.658       2651          35   \n",
       "4                                              0.755       2418          20   \n",
       "\n",
       "   # Unique Peptides  ...  Abundances (Normalized): F4: Sample, Etoposide  \\\n",
       "0                 65  ...                                    1.194626e+10   \n",
       "1                268  ...                                    1.772634e+09   \n",
       "2                  1  ...                                    7.444967e+09   \n",
       "3                  1  ...                                    1.120958e+06   \n",
       "4                 20  ...                                    1.199184e+09   \n",
       "\n",
       "   Abundances (Normalized): F5: Sample, Etoposide  \\\n",
       "0                                    8.138632e+09   \n",
       "1                                    2.325231e+09   \n",
       "2                                    5.882621e+09   \n",
       "3                                    1.139723e+06   \n",
       "4                                    8.301472e+08   \n",
       "\n",
       "  Abundances (Normalized): F6: Sample, Etoposide  \\\n",
       "0                                   8.359638e+09   \n",
       "1                                   2.399826e+09   \n",
       "2                                   5.694489e+09   \n",
       "3                                   1.601739e+06   \n",
       "4                                   9.464704e+08   \n",
       "\n",
       "   Abundances (Normalized): F10: Sample, FTY720  \\\n",
       "0                                  1.037863e+10   \n",
       "1                                  2.122741e+09   \n",
       "2                                  5.942383e+09   \n",
       "3                                  3.598291e+06   \n",
       "4                                  8.816873e+08   \n",
       "\n",
       "   Abundances (Normalized): F11: Sample, FTY720  \\\n",
       "0                                  1.260664e+10   \n",
       "1                                  2.138561e+09   \n",
       "2                                  7.232767e+09   \n",
       "3                                  1.953038e+06   \n",
       "4                                  9.276548e+08   \n",
       "\n",
       "   Abundances (Normalized): F12: Sample, FTY720  \\\n",
       "0                                  1.303859e+10   \n",
       "1                                  2.067453e+09   \n",
       "2                                  6.858702e+09   \n",
       "3                                  6.898486e+05   \n",
       "4                                  1.222988e+09   \n",
       "\n",
       "   Abundances (Normalized): F7: Sample, FTY720_plus_Etoposide  \\\n",
       "0                                       1.286167e+10            \n",
       "1                                       1.941775e+09            \n",
       "2                                       6.464013e+09            \n",
       "3                                       1.758126e+06            \n",
       "4                                       1.461613e+09            \n",
       "\n",
       "   Abundances (Normalized): F8: Sample, FTY720_plus_Etoposide  \\\n",
       "0                                       1.279744e+10            \n",
       "1                                       2.051846e+09            \n",
       "2                                       6.143735e+09            \n",
       "3                                       1.617831e+06            \n",
       "4                                       1.278716e+09            \n",
       "\n",
       "   Abundances (Normalized): F9: Sample, FTY720_plus_Etoposide  Contaminant  \n",
       "0                                       1.327241e+10                 False  \n",
       "1                                       1.983623e+09                 False  \n",
       "2                                       6.091506e+09                  True  \n",
       "3                                       1.715513e+06                 False  \n",
       "4                                       1.370481e+09                 False  \n",
       "\n",
       "[5 rows x 38 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteome.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "cd668082",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Master', 'Accession', 'Description', 'Coverage [%]',\n",
       "       'Abundance Ratio: (Etoposide) / (Control)',\n",
       "       'Abundance Ratio: (FTY720) / (Control)',\n",
       "       'Abundance Ratio: (FTY720_plus_Etoposide) / (Control)', '# PSMs',\n",
       "       '# Peptides', '# Unique Peptides', '# Protein Unique Peptides',\n",
       "       'MW [kDa]', 'Gene Symbol', 'Abundance: F1: Sample, Control',\n",
       "       'Abundance: F2: Sample, Control', 'Abundance: F3: Sample, Control',\n",
       "       'Abundance: F4: Sample, Etoposide', 'Abundance: F5: Sample, Etoposide',\n",
       "       'Abundance: F6: Sample, Etoposide', 'Abundance: F10: Sample, FTY720',\n",
       "       'Abundance: F11: Sample, FTY720', 'Abundance: F12: Sample, FTY720',\n",
       "       'Abundance: F7: Sample, FTY720_plus_Etoposide',\n",
       "       'Abundance: F8: Sample, FTY720_plus_Etoposide',\n",
       "       'Abundance: F9: Sample, FTY720_plus_Etoposide',\n",
       "       'Abundances (Normalized): F1: Sample, Control',\n",
       "       'Abundances (Normalized): F2: Sample, Control',\n",
       "       'Abundances (Normalized): F3: Sample, Control',\n",
       "       'Abundances (Normalized): F4: Sample, Etoposide',\n",
       "       'Abundances (Normalized): F5: Sample, Etoposide',\n",
       "       'Abundances (Normalized): F6: Sample, Etoposide',\n",
       "       'Abundances (Normalized): F10: Sample, FTY720',\n",
       "       'Abundances (Normalized): F11: Sample, FTY720',\n",
       "       'Abundances (Normalized): F12: Sample, FTY720',\n",
       "       'Abundances (Normalized): F7: Sample, FTY720_plus_Etoposide',\n",
       "       'Abundances (Normalized): F8: Sample, FTY720_plus_Etoposide',\n",
       "       'Abundances (Normalized): F9: Sample, FTY720_plus_Etoposide',\n",
       "       'Contaminant'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteome.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4ec97b2b",
   "metadata": {},
   "outputs": [],
   "source": [
    "control_cols = [\"Abundances (Normalized): F1: Sample, Control\", \"Abundances (Normalized): F2: Sample, Control\", \"Abundances (Normalized): F3: Sample, Control\"]\n",
    "etoposide_cols = [\"Abundances (Normalized): F4: Sample, Etoposide\", \"Abundances (Normalized): F5: Sample, Etoposide\", \"Abundances (Normalized): F6: Sample, Etoposide\"]\n",
    "fty_cols = [\"Abundances (Normalized): F10: Sample, FTY720\", \"Abundances (Normalized): F11: Sample, FTY720\", \"Abundances (Normalized): F12: Sample, FTY720\"]\n",
    "fty_eto_cols = [\"Abundances (Normalized): F7: Sample, FTY720_plus_Etoposide\", \"Abundances (Normalized): F8: Sample, FTY720_plus_Etoposide\",\"Abundances (Normalized): F9: Sample, FTY720_plus_Etoposide\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4f49794f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_rows(row):\n",
    "    control_nan_count = row[control_cols].isna().sum()\n",
    "    eoposide_nan_count = row[etoposide_cols].isna().sum()\n",
    "    fty_nan_count = row[fty_cols].isna().sum()\n",
    "    ftyeto_nan_count = row[fty_eto_cols].isna().sum()\n",
    "    return not ((control_nan_count > 1) or (eoposide_nan_count > 1) or (fty_nan_count > 1) or (ftyeto_nan_count > 1)) \n",
    "\n",
    "filtered_proteome = proteome[proteome.apply(filter_rows, axis=1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "2be6ee88",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(4525, 38)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Master                                                           0\n",
       "Accession                                                        0\n",
       "Description                                                      0\n",
       "Coverage [%]                                                     0\n",
       "Abundance Ratio: (Etoposide) / (Control)                       825\n",
       "Abundance Ratio: (FTY720) / (Control)                          774\n",
       "Abundance Ratio: (FTY720_plus_Etoposide) / (Control)           817\n",
       "# PSMs                                                           0\n",
       "# Peptides                                                       0\n",
       "# Unique Peptides                                                0\n",
       "# Protein Unique Peptides                                        0\n",
       "MW [kDa]                                                         0\n",
       "Gene Symbol                                                     32\n",
       "Abundance: F1: Sample, Control                                1099\n",
       "Abundance: F2: Sample, Control                                1102\n",
       "Abundance: F3: Sample, Control                                1098\n",
       "Abundance: F4: Sample, Etoposide                              1047\n",
       "Abundance: F5: Sample, Etoposide                              1090\n",
       "Abundance: F6: Sample, Etoposide                              1106\n",
       "Abundance: F10: Sample, FTY720                                1250\n",
       "Abundance: F11: Sample, FTY720                                1406\n",
       "Abundance: F12: Sample, FTY720                                1168\n",
       "Abundance: F7: Sample, FTY720_plus_Etoposide                  1004\n",
       "Abundance: F8: Sample, FTY720_plus_Etoposide                   990\n",
       "Abundance: F9: Sample, FTY720_plus_Etoposide                   987\n",
       "Abundances (Normalized): F1: Sample, Control                  1099\n",
       "Abundances (Normalized): F2: Sample, Control                  1102\n",
       "Abundances (Normalized): F3: Sample, Control                  1098\n",
       "Abundances (Normalized): F4: Sample, Etoposide                1047\n",
       "Abundances (Normalized): F5: Sample, Etoposide                1090\n",
       "Abundances (Normalized): F6: Sample, Etoposide                1106\n",
       "Abundances (Normalized): F10: Sample, FTY720                  1250\n",
       "Abundances (Normalized): F11: Sample, FTY720                  1406\n",
       "Abundances (Normalized): F12: Sample, FTY720                  1168\n",
       "Abundances (Normalized): F7: Sample, FTY720_plus_Etoposide    1004\n",
       "Abundances (Normalized): F8: Sample, FTY720_plus_Etoposide     990\n",
       "Abundances (Normalized): F9: Sample, FTY720_plus_Etoposide     987\n",
       "Contaminant                                                      0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(proteome.shape)\n",
    "proteome.isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "33802715",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2928, 38)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Master                                                          0\n",
       "Accession                                                       0\n",
       "Description                                                     0\n",
       "Coverage [%]                                                    0\n",
       "Abundance Ratio: (Etoposide) / (Control)                       95\n",
       "Abundance Ratio: (FTY720) / (Control)                          23\n",
       "Abundance Ratio: (FTY720_plus_Etoposide) / (Control)           88\n",
       "# PSMs                                                          0\n",
       "# Peptides                                                      0\n",
       "# Unique Peptides                                               0\n",
       "# Protein Unique Peptides                                       0\n",
       "MW [kDa]                                                        0\n",
       "Gene Symbol                                                    17\n",
       "Abundance: F1: Sample, Control                                 20\n",
       "Abundance: F2: Sample, Control                                 18\n",
       "Abundance: F3: Sample, Control                                 19\n",
       "Abundance: F4: Sample, Etoposide                               23\n",
       "Abundance: F5: Sample, Etoposide                               31\n",
       "Abundance: F6: Sample, Etoposide                               23\n",
       "Abundance: F10: Sample, FTY720                                 76\n",
       "Abundance: F11: Sample, FTY720                                170\n",
       "Abundance: F12: Sample, FTY720                                 55\n",
       "Abundance: F7: Sample, FTY720_plus_Etoposide                   11\n",
       "Abundance: F8: Sample, FTY720_plus_Etoposide                    9\n",
       "Abundance: F9: Sample, FTY720_plus_Etoposide                    9\n",
       "Abundances (Normalized): F1: Sample, Control                   20\n",
       "Abundances (Normalized): F2: Sample, Control                   18\n",
       "Abundances (Normalized): F3: Sample, Control                   19\n",
       "Abundances (Normalized): F4: Sample, Etoposide                 23\n",
       "Abundances (Normalized): F5: Sample, Etoposide                 31\n",
       "Abundances (Normalized): F6: Sample, Etoposide                 23\n",
       "Abundances (Normalized): F10: Sample, FTY720                   76\n",
       "Abundances (Normalized): F11: Sample, FTY720                  170\n",
       "Abundances (Normalized): F12: Sample, FTY720                   55\n",
       "Abundances (Normalized): F7: Sample, FTY720_plus_Etoposide     11\n",
       "Abundances (Normalized): F8: Sample, FTY720_plus_Etoposide      9\n",
       "Abundances (Normalized): F9: Sample, FTY720_plus_Etoposide      9\n",
       "Contaminant                                                     0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(filtered_proteome.shape)\n",
    "filtered_proteome.isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5092a747",
   "metadata": {},
   "outputs": [],
   "source": [
    "def impute_group(group):\n",
    "    imputer = IterativeImputer(estimator=RandomForestRegressor(), max_iter=10, random_state=0)\n",
    "    imputed_group = imputer.fit_transform(group)\n",
    "    return pd.DataFrame(imputed_group, columns=group.columns, index=group.index)\n",
    "\n",
    "control_imputed = impute_group(filtered_proteome[control_cols])\n",
    "filtered_proteome[control_cols] = control_imputed\n",
    "\n",
    "etoposide_imputed = impute_group(filtered_proteome[etoposide_cols])\n",
    "filtered_proteome[etoposide_cols] = etoposide_imputed\n",
    "\n",
    "fty_imputed = impute_group(filtered_proteome[fty_cols])\n",
    "filtered_proteome[fty_cols] = fty_imputed\n",
    "\n",
    "fty_eto_imputed = impute_group(filtered_proteome[fty_eto_cols])\n",
    "filtered_proteome[fty_eto_cols] = fty_eto_imputed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6209336b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Master                                                          0\n",
      "Accession                                                       0\n",
      "Description                                                     0\n",
      "Coverage [%]                                                    0\n",
      "Abundance Ratio: (Etoposide) / (Control)                       95\n",
      "Abundance Ratio: (FTY720) / (Control)                          23\n",
      "Abundance Ratio: (FTY720_plus_Etoposide) / (Control)           88\n",
      "# PSMs                                                          0\n",
      "# Peptides                                                      0\n",
      "# Unique Peptides                                               0\n",
      "# Protein Unique Peptides                                       0\n",
      "MW [kDa]                                                        0\n",
      "Gene Symbol                                                    17\n",
      "Abundance: F1: Sample, Control                                 20\n",
      "Abundance: F2: Sample, Control                                 18\n",
      "Abundance: F3: Sample, Control                                 19\n",
      "Abundance: F4: Sample, Etoposide                               23\n",
      "Abundance: F5: Sample, Etoposide                               31\n",
      "Abundance: F6: Sample, Etoposide                               23\n",
      "Abundance: F10: Sample, FTY720                                 76\n",
      "Abundance: F11: Sample, FTY720                                170\n",
      "Abundance: F12: Sample, FTY720                                 55\n",
      "Abundance: F7: Sample, FTY720_plus_Etoposide                   11\n",
      "Abundance: F8: Sample, FTY720_plus_Etoposide                    9\n",
      "Abundance: F9: Sample, FTY720_plus_Etoposide                    9\n",
      "Abundances (Normalized): F1: Sample, Control                    0\n",
      "Abundances (Normalized): F2: Sample, Control                    0\n",
      "Abundances (Normalized): F3: Sample, Control                    0\n",
      "Abundances (Normalized): F4: Sample, Etoposide                  0\n",
      "Abundances (Normalized): F5: Sample, Etoposide                  0\n",
      "Abundances (Normalized): F6: Sample, Etoposide                  0\n",
      "Abundances (Normalized): F10: Sample, FTY720                    0\n",
      "Abundances (Normalized): F11: Sample, FTY720                    0\n",
      "Abundances (Normalized): F12: Sample, FTY720                    0\n",
      "Abundances (Normalized): F7: Sample, FTY720_plus_Etoposide      0\n",
      "Abundances (Normalized): F8: Sample, FTY720_plus_Etoposide      0\n",
      "Abundances (Normalized): F9: Sample, FTY720_plus_Etoposide      0\n",
      "Contaminant                                                     0\n",
      "dtype: int64\n",
      "(2928, 38)\n"
     ]
    }
   ],
   "source": [
    "print(filtered_proteome.isna().sum())\n",
    "print(filtered_proteome.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "4ebd4530",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Master', 'Accession', 'Description', 'Coverage [%]',\n",
       "       'Abundance Ratio: (Etoposide) / (Control)',\n",
       "       'Abundance Ratio: (FTY720) / (Control)',\n",
       "       'Abundance Ratio: (FTY720_plus_Etoposide) / (Control)', '# PSMs',\n",
       "       '# Peptides', '# Unique Peptides', '# Protein Unique Peptides',\n",
       "       'MW [kDa]', 'Gene Symbol', 'Abundance: F1: Sample, Control',\n",
       "       'Abundance: F2: Sample, Control', 'Abundance: F3: Sample, Control',\n",
       "       'Abundance: F4: Sample, Etoposide', 'Abundance: F5: Sample, Etoposide',\n",
       "       'Abundance: F6: Sample, Etoposide', 'Abundance: F10: Sample, FTY720',\n",
       "       'Abundance: F11: Sample, FTY720', 'Abundance: F12: Sample, FTY720',\n",
       "       'Abundance: F7: Sample, FTY720_plus_Etoposide',\n",
       "       'Abundance: F8: Sample, FTY720_plus_Etoposide',\n",
       "       'Abundance: F9: Sample, FTY720_plus_Etoposide',\n",
       "       'Abundances (Normalized): F1: Sample, Control',\n",
       "       'Abundances (Normalized): F2: Sample, Control',\n",
       "       'Abundances (Normalized): F3: Sample, Control',\n",
       "       'Abundances (Normalized): F4: Sample, Etoposide',\n",
       "       'Abundances (Normalized): F5: Sample, Etoposide',\n",
       "       'Abundances (Normalized): F6: Sample, Etoposide',\n",
       "       'Abundances (Normalized): F10: Sample, FTY720',\n",
       "       'Abundances (Normalized): F11: Sample, FTY720',\n",
       "       'Abundances (Normalized): F12: Sample, FTY720',\n",
       "       'Abundances (Normalized): F7: Sample, FTY720_plus_Etoposide',\n",
       "       'Abundances (Normalized): F8: Sample, FTY720_plus_Etoposide',\n",
       "       'Abundances (Normalized): F9: Sample, FTY720_plus_Etoposide',\n",
       "       'Contaminant'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_proteome.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "fc9864d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "proteome = filtered_proteome.loc[:, [\n",
    "    \"Gene Symbol\",\n",
    "    'Abundances (Normalized): F1: Sample, Control',\n",
    "    'Abundances (Normalized): F2: Sample, Control',\n",
    "    'Abundances (Normalized): F3: Sample, Control',\n",
    "    'Abundances (Normalized): F4: Sample, Etoposide',\n",
    "    'Abundances (Normalized): F5: Sample, Etoposide',\n",
    "    'Abundances (Normalized): F6: Sample, Etoposide',\n",
    "    'Abundances (Normalized): F10: Sample, FTY720',\n",
    "    'Abundances (Normalized): F11: Sample, FTY720',\n",
    "    'Abundances (Normalized): F12: Sample, FTY720',\n",
    "    'Abundances (Normalized): F7: Sample, FTY720_plus_Etoposide',\n",
    "    'Abundances (Normalized): F8: Sample, FTY720_plus_Etoposide',\n",
    "    'Abundances (Normalized): F9: Sample, FTY720_plus_Etoposide',\n",
    "    'Contaminant'\n",
    "]]\n",
    "\n",
    "# columns_to_select = [\n",
    "#    \"Gene Symbol\",\n",
    "#    'Abundances (Normalized): F1: Sample, Control',\n",
    "#    'Abundances (Normalized): F2: Sample, Control',\n",
    "#    'Abundances (Normalized): F3: Sample, Control',\n",
    "#    'Abundances (Normalized): F4: Sample, Etoposide',\n",
    "#    'Abundances (Normalized): F5: Sample, Etoposide',\n",
    "#    'Abundances (Normalized): F6: Sample, Etoposide',\n",
    "#    'Abundances (Normalized): F10: Sample, FTY720',\n",
    "#    'Abundances (Normalized): F11: Sample, FTY720',\n",
    "#    'Abundances (Normalized): F12: Sample, FTY720',\n",
    "#    'Abundances (Normalized): F7: Sample, FTY720_plus_Etoposide',\n",
    "#    'Abundances (Normalized): F8: Sample, FTY720_plus_Etoposide',\n",
    "#    'Abundances (Normalized): F9: Sample, FTY720_plus_Etoposide',\n",
    "#   'Contaminant']\n",
    "\n",
    "# proteome = filtered_proteome[columns_to_select]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f0214086",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene Symbol                                                   17\n",
      "Abundances (Normalized): F1: Sample, Control                   0\n",
      "Abundances (Normalized): F2: Sample, Control                   0\n",
      "Abundances (Normalized): F3: Sample, Control                   0\n",
      "Abundances (Normalized): F4: Sample, Etoposide                 0\n",
      "Abundances (Normalized): F5: Sample, Etoposide                 0\n",
      "Abundances (Normalized): F6: Sample, Etoposide                 0\n",
      "Abundances (Normalized): F10: Sample, FTY720                   0\n",
      "Abundances (Normalized): F11: Sample, FTY720                   0\n",
      "Abundances (Normalized): F12: Sample, FTY720                   0\n",
      "Abundances (Normalized): F7: Sample, FTY720_plus_Etoposide     0\n",
      "Abundances (Normalized): F8: Sample, FTY720_plus_Etoposide     0\n",
      "Abundances (Normalized): F9: Sample, FTY720_plus_Etoposide     0\n",
      "Contaminant                                                    0\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print(proteome.isna().sum())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "68ca91ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "proteome = proteome.dropna(subset=[\"Gene Symbol\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "0b8dc915",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2911, 14)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteome.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "49954936",
   "metadata": {},
   "outputs": [],
   "source": [
    "proteome['Mean_CTRL'] = gmean(proteome[control_cols], axis=1)\n",
    "proteome['Mean_Eto'] = gmean(proteome[etoposide_cols], axis=1)\n",
    "proteome['Mean_FTY'] = gmean(proteome[fty_cols], axis=1)\n",
    "proteome['Mean_FTY_Eto'] = gmean(proteome[fty_eto_cols], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "c878abc5",
   "metadata": {},
   "outputs": [],
   "source": [
    "proteome[\"FC_Eto/CTRL\"] = proteome['Mean_Eto']/proteome['Mean_CTRL']\n",
    "proteome['log2_FC_Eto/CTRL'] = np.log2(proteome['FC_Eto/CTRL'])\n",
    "\n",
    "proteome[\"FC_FTY/CTRL\"] = proteome['Mean_FTY']/proteome['Mean_CTRL']\n",
    "proteome['log2_FC_FTY/CTRL'] = np.log2(proteome['FC_FTY/CTRL'])\n",
    "\n",
    "proteome[\"FC_FTY_Eto/CTRL\"] = proteome['Mean_FTY_Eto']/proteome['Mean_CTRL']\n",
    "proteome['log2_FC_FTY_Eto/CTRL'] = np.log2(proteome['FC_FTY_Eto/CTRL'])\n",
    "\n",
    "proteome[\"FC_FTY/Eto\"] = proteome['Mean_FTY']/proteome['Mean_Eto']\n",
    "proteome['log2_FC_FTY/Eto'] = np.log2(proteome['FC_FTY/Eto'])\n",
    "\n",
    "proteome[\"FC_FTY_Eto/Eto\"] = proteome['Mean_FTY_Eto']/proteome['Mean_Eto']\n",
    "proteome['log2_FC_FTY_Eto/Eto'] = np.log2(proteome['FC_FTY_Eto/Eto'])\n",
    "\n",
    "proteome[\"FC_FTY_Eto/FTY\"] = proteome['Mean_FTY_Eto']/proteome['Mean_FTY']\n",
    "proteome['log2_FC_FTY_Eto/FTY'] = np.log2(proteome['FC_FTY_Eto/FTY'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "3b6efdfb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>Abundances (Normalized): F1: Sample, Control</th>\n",
       "      <th>Abundances (Normalized): F2: Sample, Control</th>\n",
       "      <th>Abundances (Normalized): F3: Sample, Control</th>\n",
       "      <th>Abundances (Normalized): F4: Sample, Etoposide</th>\n",
       "      <th>Abundances (Normalized): F5: Sample, Etoposide</th>\n",
       "      <th>Abundances (Normalized): F6: Sample, Etoposide</th>\n",
       "      <th>Abundances (Normalized): F10: Sample, FTY720</th>\n",
       "      <th>Abundances (Normalized): F11: Sample, FTY720</th>\n",
       "      <th>Abundances (Normalized): F12: Sample, FTY720</th>\n",
       "      <th>...</th>\n",
       "      <th>FC_FTY/CTRL</th>\n",
       "      <th>log2_FC_FTY/CTRL</th>\n",
       "      <th>FC_FTY_Eto/CTRL</th>\n",
       "      <th>log2_FC_FTY_Eto/CTRL</th>\n",
       "      <th>FC_FTY/Eto</th>\n",
       "      <th>log2_FC_FTY/Eto</th>\n",
       "      <th>FC_FTY_Eto/Eto</th>\n",
       "      <th>log2_FC_FTY_Eto/Eto</th>\n",
       "      <th>FC_FTY_Eto/FTY</th>\n",
       "      <th>log2_FC_FTY_Eto/FTY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>VIM</td>\n",
       "      <td>1.649179e+10</td>\n",
       "      <td>1.661164e+10</td>\n",
       "      <td>1.524631e+10</td>\n",
       "      <td>1.194626e+10</td>\n",
       "      <td>8.138632e+09</td>\n",
       "      <td>8.359638e+09</td>\n",
       "      <td>1.037863e+10</td>\n",
       "      <td>1.260664e+10</td>\n",
       "      <td>1.303859e+10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.741950</td>\n",
       "      <td>-0.430605</td>\n",
       "      <td>0.805703</td>\n",
       "      <td>-0.311680</td>\n",
       "      <td>1.280363</td>\n",
       "      <td>0.356552</td>\n",
       "      <td>1.390379</td>\n",
       "      <td>0.475478</td>\n",
       "      <td>1.085926</td>\n",
       "      <td>0.118926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AHNAK</td>\n",
       "      <td>2.003021e+09</td>\n",
       "      <td>1.849071e+09</td>\n",
       "      <td>1.713771e+09</td>\n",
       "      <td>1.772634e+09</td>\n",
       "      <td>2.325231e+09</td>\n",
       "      <td>2.399826e+09</td>\n",
       "      <td>2.122741e+09</td>\n",
       "      <td>2.138561e+09</td>\n",
       "      <td>2.067453e+09</td>\n",
       "      <td>...</td>\n",
       "      <td>1.139255</td>\n",
       "      <td>0.188090</td>\n",
       "      <td>1.075813</td>\n",
       "      <td>0.105428</td>\n",
       "      <td>0.982645</td>\n",
       "      <td>-0.025258</td>\n",
       "      <td>0.927924</td>\n",
       "      <td>-0.107921</td>\n",
       "      <td>0.944313</td>\n",
       "      <td>-0.082663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTB</td>\n",
       "      <td>5.672231e+09</td>\n",
       "      <td>5.636443e+09</td>\n",
       "      <td>5.878443e+09</td>\n",
       "      <td>7.444967e+09</td>\n",
       "      <td>5.882621e+09</td>\n",
       "      <td>5.694489e+09</td>\n",
       "      <td>5.942383e+09</td>\n",
       "      <td>7.232767e+09</td>\n",
       "      <td>6.858702e+09</td>\n",
       "      <td>...</td>\n",
       "      <td>1.161882</td>\n",
       "      <td>0.216463</td>\n",
       "      <td>1.087792</td>\n",
       "      <td>0.121403</td>\n",
       "      <td>1.057319</td>\n",
       "      <td>0.080411</td>\n",
       "      <td>0.989898</td>\n",
       "      <td>-0.014649</td>\n",
       "      <td>0.936233</td>\n",
       "      <td>-0.095060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACTG1</td>\n",
       "      <td>2.238417e+06</td>\n",
       "      <td>2.190261e+06</td>\n",
       "      <td>2.738121e+06</td>\n",
       "      <td>1.120958e+06</td>\n",
       "      <td>1.139723e+06</td>\n",
       "      <td>1.601739e+06</td>\n",
       "      <td>3.598291e+06</td>\n",
       "      <td>1.953038e+06</td>\n",
       "      <td>6.898486e+05</td>\n",
       "      <td>...</td>\n",
       "      <td>0.712126</td>\n",
       "      <td>-0.489795</td>\n",
       "      <td>0.713667</td>\n",
       "      <td>-0.486676</td>\n",
       "      <td>1.333092</td>\n",
       "      <td>0.414776</td>\n",
       "      <td>1.335976</td>\n",
       "      <td>0.417895</td>\n",
       "      <td>1.002164</td>\n",
       "      <td>0.003118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GAPDH</td>\n",
       "      <td>1.242975e+09</td>\n",
       "      <td>1.282440e+09</td>\n",
       "      <td>1.185116e+09</td>\n",
       "      <td>1.199184e+09</td>\n",
       "      <td>8.301472e+08</td>\n",
       "      <td>9.464704e+08</td>\n",
       "      <td>8.816873e+08</td>\n",
       "      <td>9.276548e+08</td>\n",
       "      <td>1.222988e+09</td>\n",
       "      <td>...</td>\n",
       "      <td>0.809011</td>\n",
       "      <td>-0.305769</td>\n",
       "      <td>1.106810</td>\n",
       "      <td>0.146408</td>\n",
       "      <td>1.020137</td>\n",
       "      <td>0.028763</td>\n",
       "      <td>1.395653</td>\n",
       "      <td>0.480940</td>\n",
       "      <td>1.368103</td>\n",
       "      <td>0.452177</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Gene Symbol  Abundances (Normalized): F1: Sample, Control  \\\n",
       "0         VIM                                  1.649179e+10   \n",
       "1       AHNAK                                  2.003021e+09   \n",
       "2        ACTB                                  5.672231e+09   \n",
       "3       ACTG1                                  2.238417e+06   \n",
       "4       GAPDH                                  1.242975e+09   \n",
       "\n",
       "   Abundances (Normalized): F2: Sample, Control  \\\n",
       "0                                  1.661164e+10   \n",
       "1                                  1.849071e+09   \n",
       "2                                  5.636443e+09   \n",
       "3                                  2.190261e+06   \n",
       "4                                  1.282440e+09   \n",
       "\n",
       "   Abundances (Normalized): F3: Sample, Control  \\\n",
       "0                                  1.524631e+10   \n",
       "1                                  1.713771e+09   \n",
       "2                                  5.878443e+09   \n",
       "3                                  2.738121e+06   \n",
       "4                                  1.185116e+09   \n",
       "\n",
       "   Abundances (Normalized): F4: Sample, Etoposide  \\\n",
       "0                                    1.194626e+10   \n",
       "1                                    1.772634e+09   \n",
       "2                                    7.444967e+09   \n",
       "3                                    1.120958e+06   \n",
       "4                                    1.199184e+09   \n",
       "\n",
       "   Abundances (Normalized): F5: Sample, Etoposide  \\\n",
       "0                                    8.138632e+09   \n",
       "1                                    2.325231e+09   \n",
       "2                                    5.882621e+09   \n",
       "3                                    1.139723e+06   \n",
       "4                                    8.301472e+08   \n",
       "\n",
       "   Abundances (Normalized): F6: Sample, Etoposide  \\\n",
       "0                                    8.359638e+09   \n",
       "1                                    2.399826e+09   \n",
       "2                                    5.694489e+09   \n",
       "3                                    1.601739e+06   \n",
       "4                                    9.464704e+08   \n",
       "\n",
       "   Abundances (Normalized): F10: Sample, FTY720  \\\n",
       "0                                  1.037863e+10   \n",
       "1                                  2.122741e+09   \n",
       "2                                  5.942383e+09   \n",
       "3                                  3.598291e+06   \n",
       "4                                  8.816873e+08   \n",
       "\n",
       "   Abundances (Normalized): F11: Sample, FTY720  \\\n",
       "0                                  1.260664e+10   \n",
       "1                                  2.138561e+09   \n",
       "2                                  7.232767e+09   \n",
       "3                                  1.953038e+06   \n",
       "4                                  9.276548e+08   \n",
       "\n",
       "   Abundances (Normalized): F12: Sample, FTY720  ...  FC_FTY/CTRL  \\\n",
       "0                                  1.303859e+10  ...     0.741950   \n",
       "1                                  2.067453e+09  ...     1.139255   \n",
       "2                                  6.858702e+09  ...     1.161882   \n",
       "3                                  6.898486e+05  ...     0.712126   \n",
       "4                                  1.222988e+09  ...     0.809011   \n",
       "\n",
       "   log2_FC_FTY/CTRL  FC_FTY_Eto/CTRL  log2_FC_FTY_Eto/CTRL  FC_FTY/Eto  \\\n",
       "0         -0.430605         0.805703             -0.311680    1.280363   \n",
       "1          0.188090         1.075813              0.105428    0.982645   \n",
       "2          0.216463         1.087792              0.121403    1.057319   \n",
       "3         -0.489795         0.713667             -0.486676    1.333092   \n",
       "4         -0.305769         1.106810              0.146408    1.020137   \n",
       "\n",
       "   log2_FC_FTY/Eto  FC_FTY_Eto/Eto  log2_FC_FTY_Eto/Eto  FC_FTY_Eto/FTY  \\\n",
       "0         0.356552        1.390379             0.475478        1.085926   \n",
       "1        -0.025258        0.927924            -0.107921        0.944313   \n",
       "2         0.080411        0.989898            -0.014649        0.936233   \n",
       "3         0.414776        1.335976             0.417895        1.002164   \n",
       "4         0.028763        1.395653             0.480940        1.368103   \n",
       "\n",
       "   log2_FC_FTY_Eto/FTY  \n",
       "0             0.118926  \n",
       "1            -0.082663  \n",
       "2            -0.095060  \n",
       "3             0.003118  \n",
       "4             0.452177  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteome.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "fc7ff82c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   Abundances (Normalized): F1: Sample, Control  \\\n",
      "0                                  1.649179e+10   \n",
      "1                                  2.003021e+09   \n",
      "2                                  5.672231e+09   \n",
      "3                                  2.238417e+06   \n",
      "4                                  1.242975e+09   \n",
      "\n",
      "   Abundances (Normalized): F2: Sample, Control  \\\n",
      "0                                  1.661164e+10   \n",
      "1                                  1.849071e+09   \n",
      "2                                  5.636443e+09   \n",
      "3                                  2.190261e+06   \n",
      "4                                  1.282440e+09   \n",
      "\n",
      "   Abundances (Normalized): F3: Sample, Control  \\\n",
      "0                                  1.524631e+10   \n",
      "1                                  1.713771e+09   \n",
      "2                                  5.878443e+09   \n",
      "3                                  2.738121e+06   \n",
      "4                                  1.185116e+09   \n",
      "\n",
      "   Abundances (Normalized): F4: Sample, Etoposide  \\\n",
      "0                                    1.194626e+10   \n",
      "1                                    1.772634e+09   \n",
      "2                                    7.444967e+09   \n",
      "3                                    1.120958e+06   \n",
      "4                                    1.199184e+09   \n",
      "\n",
      "   Abundances (Normalized): F5: Sample, Etoposide  \\\n",
      "0                                    8.138632e+09   \n",
      "1                                    2.325231e+09   \n",
      "2                                    5.882621e+09   \n",
      "3                                    1.139723e+06   \n",
      "4                                    8.301472e+08   \n",
      "\n",
      "   Abundances (Normalized): F6: Sample, Etoposide  \\\n",
      "0                                    8.359638e+09   \n",
      "1                                    2.399826e+09   \n",
      "2                                    5.694489e+09   \n",
      "3                                    1.601739e+06   \n",
      "4                                    9.464704e+08   \n",
      "\n",
      "   Abundances (Normalized): F10: Sample, FTY720  \\\n",
      "0                                  1.037863e+10   \n",
      "1                                  2.122741e+09   \n",
      "2                                  5.942383e+09   \n",
      "3                                  3.598291e+06   \n",
      "4                                  8.816873e+08   \n",
      "\n",
      "   Abundances (Normalized): F11: Sample, FTY720  \\\n",
      "0                                  1.260664e+10   \n",
      "1                                  2.138561e+09   \n",
      "2                                  7.232767e+09   \n",
      "3                                  1.953038e+06   \n",
      "4                                  9.276548e+08   \n",
      "\n",
      "   Abundances (Normalized): F12: Sample, FTY720  \n",
      "0                                  1.303859e+10  \n",
      "1                                  2.067453e+09  \n",
      "2                                  6.858702e+09  \n",
      "3                                  6.898486e+05  \n",
      "4                                  1.222988e+09  \n"
     ]
    }
   ],
   "source": [
    "expression_data = proteome.iloc[:, 1:10]\n",
    "genes = proteome.iloc[:, 0]\n",
    "print(expression_data.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "8092288d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                          F_value                   p_value\n",
      "Gene Symbol                                                \n",
      "VIM          [14.061796499856634]     [0.00543612570756712]\n",
      "AHNAK        [1.7640375647415338]     [0.24971132113926583]\n",
      "ACTB         [1.5076497826418662]     [0.29479035108718127]\n",
      "ACTG1         [1.267140157952302]     [0.34749858502481396]\n",
      "GAPDH        [2.3121848680121535]     [0.18011265286899675]\n",
      "...                           ...                       ...\n",
      "TRAPPC2L     [215.27665626992496]  [2.5962223676529114e-06]\n",
      "BICC1         [8.121658433351493]    [0.019627053454329406]\n",
      "OSBPL1A      [3.3089279713994175]      [0.1075219313609997]\n",
      "ATXN3         [3.854482605582061]     [0.08383777195930091]\n",
      "MECP2         [2.574457760566768]      [0.1558676103244947]\n",
      "\n",
      "[2911 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "expression_data = proteome.iloc[:, 1:10]\n",
    "genes = proteome.iloc[:, 0]\n",
    "\n",
    "\n",
    "groups = [expression_data.iloc[:, i:i+3] for i in range(0, len(expression_data.columns), 3)]\n",
    "\n",
    "\n",
    "P_Val = []\n",
    "for i in range(len(expression_data)):\n",
    "    data_i = pd.DataFrame(expression_data.iloc[i, :])  # Convert to DataFrame\n",
    "    group_i = [data_i.iloc[j:j+3, :] for j in range(0, len(data_i), 3)]  # Access rows for each group\n",
    "    p_value = f_oneway(*group_i)\n",
    "    P_Val.append(p_value)\n",
    "\n",
    "\n",
    "p_values_df = pd.DataFrame(P_Val, index=genes, columns=['F_value', 'p_value'])  # Adjusted to include F_value\n",
    "\n",
    "\n",
    "print(p_values_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "f78cb89c",
   "metadata": {},
   "outputs": [],
   "source": [
    "p_values_df.to_excel(\"/Users/mortezaabyadeh/Desktop/p_values.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "f80917bd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>Abundances (Normalized): F1: Sample, Control</th>\n",
       "      <th>Abundances (Normalized): F2: Sample, Control</th>\n",
       "      <th>Abundances (Normalized): F3: Sample, Control</th>\n",
       "      <th>Abundances (Normalized): F4: Sample, Etoposide</th>\n",
       "      <th>Abundances (Normalized): F5: Sample, Etoposide</th>\n",
       "      <th>Abundances (Normalized): F6: Sample, Etoposide</th>\n",
       "      <th>Abundances (Normalized): F10: Sample, FTY720</th>\n",
       "      <th>Abundances (Normalized): F11: Sample, FTY720</th>\n",
       "      <th>Abundances (Normalized): F12: Sample, FTY720</th>\n",
       "      <th>...</th>\n",
       "      <th>FC_FTY/CTRL</th>\n",
       "      <th>log2_FC_FTY/CTRL</th>\n",
       "      <th>FC_FTY_Eto/CTRL</th>\n",
       "      <th>log2_FC_FTY_Eto/CTRL</th>\n",
       "      <th>FC_FTY/Eto</th>\n",
       "      <th>log2_FC_FTY/Eto</th>\n",
       "      <th>FC_FTY_Eto/Eto</th>\n",
       "      <th>log2_FC_FTY_Eto/Eto</th>\n",
       "      <th>FC_FTY_Eto/FTY</th>\n",
       "      <th>log2_FC_FTY_Eto/FTY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>VIM</td>\n",
       "      <td>1.649179e+10</td>\n",
       "      <td>1.661164e+10</td>\n",
       "      <td>1.524631e+10</td>\n",
       "      <td>1.194626e+10</td>\n",
       "      <td>8.138632e+09</td>\n",
       "      <td>8.359638e+09</td>\n",
       "      <td>1.037863e+10</td>\n",
       "      <td>1.260664e+10</td>\n",
       "      <td>1.303859e+10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.741950</td>\n",
       "      <td>-0.430605</td>\n",
       "      <td>0.805703</td>\n",
       "      <td>-0.311680</td>\n",
       "      <td>1.280363</td>\n",
       "      <td>0.356552</td>\n",
       "      <td>1.390379</td>\n",
       "      <td>0.475478</td>\n",
       "      <td>1.085926</td>\n",
       "      <td>0.118926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AHNAK</td>\n",
       "      <td>2.003021e+09</td>\n",
       "      <td>1.849071e+09</td>\n",
       "      <td>1.713771e+09</td>\n",
       "      <td>1.772634e+09</td>\n",
       "      <td>2.325231e+09</td>\n",
       "      <td>2.399826e+09</td>\n",
       "      <td>2.122741e+09</td>\n",
       "      <td>2.138561e+09</td>\n",
       "      <td>2.067453e+09</td>\n",
       "      <td>...</td>\n",
       "      <td>1.139255</td>\n",
       "      <td>0.188090</td>\n",
       "      <td>1.075813</td>\n",
       "      <td>0.105428</td>\n",
       "      <td>0.982645</td>\n",
       "      <td>-0.025258</td>\n",
       "      <td>0.927924</td>\n",
       "      <td>-0.107921</td>\n",
       "      <td>0.944313</td>\n",
       "      <td>-0.082663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTB</td>\n",
       "      <td>5.672231e+09</td>\n",
       "      <td>5.636443e+09</td>\n",
       "      <td>5.878443e+09</td>\n",
       "      <td>7.444967e+09</td>\n",
       "      <td>5.882621e+09</td>\n",
       "      <td>5.694489e+09</td>\n",
       "      <td>5.942383e+09</td>\n",
       "      <td>7.232767e+09</td>\n",
       "      <td>6.858702e+09</td>\n",
       "      <td>...</td>\n",
       "      <td>1.161882</td>\n",
       "      <td>0.216463</td>\n",
       "      <td>1.087792</td>\n",
       "      <td>0.121403</td>\n",
       "      <td>1.057319</td>\n",
       "      <td>0.080411</td>\n",
       "      <td>0.989898</td>\n",
       "      <td>-0.014649</td>\n",
       "      <td>0.936233</td>\n",
       "      <td>-0.095060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACTG1</td>\n",
       "      <td>2.238417e+06</td>\n",
       "      <td>2.190261e+06</td>\n",
       "      <td>2.738121e+06</td>\n",
       "      <td>1.120958e+06</td>\n",
       "      <td>1.139723e+06</td>\n",
       "      <td>1.601739e+06</td>\n",
       "      <td>3.598291e+06</td>\n",
       "      <td>1.953038e+06</td>\n",
       "      <td>6.898486e+05</td>\n",
       "      <td>...</td>\n",
       "      <td>0.712126</td>\n",
       "      <td>-0.489795</td>\n",
       "      <td>0.713667</td>\n",
       "      <td>-0.486676</td>\n",
       "      <td>1.333092</td>\n",
       "      <td>0.414776</td>\n",
       "      <td>1.335976</td>\n",
       "      <td>0.417895</td>\n",
       "      <td>1.002164</td>\n",
       "      <td>0.003118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GAPDH</td>\n",
       "      <td>1.242975e+09</td>\n",
       "      <td>1.282440e+09</td>\n",
       "      <td>1.185116e+09</td>\n",
       "      <td>1.199184e+09</td>\n",
       "      <td>8.301472e+08</td>\n",
       "      <td>9.464704e+08</td>\n",
       "      <td>8.816873e+08</td>\n",
       "      <td>9.276548e+08</td>\n",
       "      <td>1.222988e+09</td>\n",
       "      <td>...</td>\n",
       "      <td>0.809011</td>\n",
       "      <td>-0.305769</td>\n",
       "      <td>1.106810</td>\n",
       "      <td>0.146408</td>\n",
       "      <td>1.020137</td>\n",
       "      <td>0.028763</td>\n",
       "      <td>1.395653</td>\n",
       "      <td>0.480940</td>\n",
       "      <td>1.368103</td>\n",
       "      <td>0.452177</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Gene Symbol  Abundances (Normalized): F1: Sample, Control  \\\n",
       "0         VIM                                  1.649179e+10   \n",
       "1       AHNAK                                  2.003021e+09   \n",
       "2        ACTB                                  5.672231e+09   \n",
       "3       ACTG1                                  2.238417e+06   \n",
       "4       GAPDH                                  1.242975e+09   \n",
       "\n",
       "   Abundances (Normalized): F2: Sample, Control  \\\n",
       "0                                  1.661164e+10   \n",
       "1                                  1.849071e+09   \n",
       "2                                  5.636443e+09   \n",
       "3                                  2.190261e+06   \n",
       "4                                  1.282440e+09   \n",
       "\n",
       "   Abundances (Normalized): F3: Sample, Control  \\\n",
       "0                                  1.524631e+10   \n",
       "1                                  1.713771e+09   \n",
       "2                                  5.878443e+09   \n",
       "3                                  2.738121e+06   \n",
       "4                                  1.185116e+09   \n",
       "\n",
       "   Abundances (Normalized): F4: Sample, Etoposide  \\\n",
       "0                                    1.194626e+10   \n",
       "1                                    1.772634e+09   \n",
       "2                                    7.444967e+09   \n",
       "3                                    1.120958e+06   \n",
       "4                                    1.199184e+09   \n",
       "\n",
       "   Abundances (Normalized): F5: Sample, Etoposide  \\\n",
       "0                                    8.138632e+09   \n",
       "1                                    2.325231e+09   \n",
       "2                                    5.882621e+09   \n",
       "3                                    1.139723e+06   \n",
       "4                                    8.301472e+08   \n",
       "\n",
       "   Abundances (Normalized): F6: Sample, Etoposide  \\\n",
       "0                                    8.359638e+09   \n",
       "1                                    2.399826e+09   \n",
       "2                                    5.694489e+09   \n",
       "3                                    1.601739e+06   \n",
       "4                                    9.464704e+08   \n",
       "\n",
       "   Abundances (Normalized): F10: Sample, FTY720  \\\n",
       "0                                  1.037863e+10   \n",
       "1                                  2.122741e+09   \n",
       "2                                  5.942383e+09   \n",
       "3                                  3.598291e+06   \n",
       "4                                  8.816873e+08   \n",
       "\n",
       "   Abundances (Normalized): F11: Sample, FTY720  \\\n",
       "0                                  1.260664e+10   \n",
       "1                                  2.138561e+09   \n",
       "2                                  7.232767e+09   \n",
       "3                                  1.953038e+06   \n",
       "4                                  9.276548e+08   \n",
       "\n",
       "   Abundances (Normalized): F12: Sample, FTY720  ...  FC_FTY/CTRL  \\\n",
       "0                                  1.303859e+10  ...     0.741950   \n",
       "1                                  2.067453e+09  ...     1.139255   \n",
       "2                                  6.858702e+09  ...     1.161882   \n",
       "3                                  6.898486e+05  ...     0.712126   \n",
       "4                                  1.222988e+09  ...     0.809011   \n",
       "\n",
       "   log2_FC_FTY/CTRL  FC_FTY_Eto/CTRL  log2_FC_FTY_Eto/CTRL  FC_FTY/Eto  \\\n",
       "0         -0.430605         0.805703             -0.311680    1.280363   \n",
       "1          0.188090         1.075813              0.105428    0.982645   \n",
       "2          0.216463         1.087792              0.121403    1.057319   \n",
       "3         -0.489795         0.713667             -0.486676    1.333092   \n",
       "4         -0.305769         1.106810              0.146408    1.020137   \n",
       "\n",
       "   log2_FC_FTY/Eto  FC_FTY_Eto/Eto  log2_FC_FTY_Eto/Eto  FC_FTY_Eto/FTY  \\\n",
       "0         0.356552        1.390379             0.475478        1.085926   \n",
       "1        -0.025258        0.927924            -0.107921        0.944313   \n",
       "2         0.080411        0.989898            -0.014649        0.936233   \n",
       "3         0.414776        1.335976             0.417895        1.002164   \n",
       "4         0.028763        1.395653             0.480940        1.368103   \n",
       "\n",
       "   log2_FC_FTY_Eto/FTY  \n",
       "0             0.118926  \n",
       "1            -0.082663  \n",
       "2            -0.095060  \n",
       "3             0.003118  \n",
       "4             0.452177  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteome.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "f750715e",
   "metadata": {},
   "outputs": [],
   "source": [
    "P = pd.read_excel(\"/Users/mortezaabyadeh/Desktop/p_values.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "9154678b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>p_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>VIM</td>\n",
       "      <td>0.005436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AHNAK</td>\n",
       "      <td>0.249711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTB</td>\n",
       "      <td>0.294790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACTG1</td>\n",
       "      <td>0.347499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GAPDH</td>\n",
       "      <td>0.180113</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Gene Symbol   p_value\n",
       "0         VIM  0.005436\n",
       "1       AHNAK  0.249711\n",
       "2        ACTB  0.294790\n",
       "3       ACTG1  0.347499\n",
       "4       GAPDH  0.180113"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "P.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "6b4acaf1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2911, 2)\n",
      "(2911, 30)\n"
     ]
    }
   ],
   "source": [
    "print(P.shape)\n",
    "print(proteome.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "cc6a8a2e",
   "metadata": {},
   "outputs": [],
   "source": [
    "proteome[\"ANOVA\"] = P[\"p_value\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "74aa39c1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>Abundances (Normalized): F1: Sample, Control</th>\n",
       "      <th>Abundances (Normalized): F2: Sample, Control</th>\n",
       "      <th>Abundances (Normalized): F3: Sample, Control</th>\n",
       "      <th>Abundances (Normalized): F4: Sample, Etoposide</th>\n",
       "      <th>Abundances (Normalized): F5: Sample, Etoposide</th>\n",
       "      <th>Abundances (Normalized): F6: Sample, Etoposide</th>\n",
       "      <th>Abundances (Normalized): F10: Sample, FTY720</th>\n",
       "      <th>Abundances (Normalized): F11: Sample, FTY720</th>\n",
       "      <th>Abundances (Normalized): F12: Sample, FTY720</th>\n",
       "      <th>...</th>\n",
       "      <th>log2_FC_FTY/CTRL</th>\n",
       "      <th>FC_FTY_Eto/CTRL</th>\n",
       "      <th>log2_FC_FTY_Eto/CTRL</th>\n",
       "      <th>FC_FTY/Eto</th>\n",
       "      <th>log2_FC_FTY/Eto</th>\n",
       "      <th>FC_FTY_Eto/Eto</th>\n",
       "      <th>log2_FC_FTY_Eto/Eto</th>\n",
       "      <th>FC_FTY_Eto/FTY</th>\n",
       "      <th>log2_FC_FTY_Eto/FTY</th>\n",
       "      <th>ANOVA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>VIM</td>\n",
       "      <td>1.649179e+10</td>\n",
       "      <td>1.661164e+10</td>\n",
       "      <td>1.524631e+10</td>\n",
       "      <td>1.194626e+10</td>\n",
       "      <td>8.138632e+09</td>\n",
       "      <td>8.359638e+09</td>\n",
       "      <td>1.037863e+10</td>\n",
       "      <td>1.260664e+10</td>\n",
       "      <td>1.303859e+10</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.430605</td>\n",
       "      <td>0.805703</td>\n",
       "      <td>-0.311680</td>\n",
       "      <td>1.280363</td>\n",
       "      <td>0.356552</td>\n",
       "      <td>1.390379</td>\n",
       "      <td>0.475478</td>\n",
       "      <td>1.085926</td>\n",
       "      <td>0.118926</td>\n",
       "      <td>0.005436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AHNAK</td>\n",
       "      <td>2.003021e+09</td>\n",
       "      <td>1.849071e+09</td>\n",
       "      <td>1.713771e+09</td>\n",
       "      <td>1.772634e+09</td>\n",
       "      <td>2.325231e+09</td>\n",
       "      <td>2.399826e+09</td>\n",
       "      <td>2.122741e+09</td>\n",
       "      <td>2.138561e+09</td>\n",
       "      <td>2.067453e+09</td>\n",
       "      <td>...</td>\n",
       "      <td>0.188090</td>\n",
       "      <td>1.075813</td>\n",
       "      <td>0.105428</td>\n",
       "      <td>0.982645</td>\n",
       "      <td>-0.025258</td>\n",
       "      <td>0.927924</td>\n",
       "      <td>-0.107921</td>\n",
       "      <td>0.944313</td>\n",
       "      <td>-0.082663</td>\n",
       "      <td>0.249711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ACTB</td>\n",
       "      <td>5.672231e+09</td>\n",
       "      <td>5.636443e+09</td>\n",
       "      <td>5.878443e+09</td>\n",
       "      <td>7.444967e+09</td>\n",
       "      <td>5.882621e+09</td>\n",
       "      <td>5.694489e+09</td>\n",
       "      <td>5.942383e+09</td>\n",
       "      <td>7.232767e+09</td>\n",
       "      <td>6.858702e+09</td>\n",
       "      <td>...</td>\n",
       "      <td>0.216463</td>\n",
       "      <td>1.087792</td>\n",
       "      <td>0.121403</td>\n",
       "      <td>1.057319</td>\n",
       "      <td>0.080411</td>\n",
       "      <td>0.989898</td>\n",
       "      <td>-0.014649</td>\n",
       "      <td>0.936233</td>\n",
       "      <td>-0.095060</td>\n",
       "      <td>0.294790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ACTG1</td>\n",
       "      <td>2.238417e+06</td>\n",
       "      <td>2.190261e+06</td>\n",
       "      <td>2.738121e+06</td>\n",
       "      <td>1.120958e+06</td>\n",
       "      <td>1.139723e+06</td>\n",
       "      <td>1.601739e+06</td>\n",
       "      <td>3.598291e+06</td>\n",
       "      <td>1.953038e+06</td>\n",
       "      <td>6.898486e+05</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.489795</td>\n",
       "      <td>0.713667</td>\n",
       "      <td>-0.486676</td>\n",
       "      <td>1.333092</td>\n",
       "      <td>0.414776</td>\n",
       "      <td>1.335976</td>\n",
       "      <td>0.417895</td>\n",
       "      <td>1.002164</td>\n",
       "      <td>0.003118</td>\n",
       "      <td>0.347499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GAPDH</td>\n",
       "      <td>1.242975e+09</td>\n",
       "      <td>1.282440e+09</td>\n",
       "      <td>1.185116e+09</td>\n",
       "      <td>1.199184e+09</td>\n",
       "      <td>8.301472e+08</td>\n",
       "      <td>9.464704e+08</td>\n",
       "      <td>8.816873e+08</td>\n",
       "      <td>9.276548e+08</td>\n",
       "      <td>1.222988e+09</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.305769</td>\n",
       "      <td>1.106810</td>\n",
       "      <td>0.146408</td>\n",
       "      <td>1.020137</td>\n",
       "      <td>0.028763</td>\n",
       "      <td>1.395653</td>\n",
       "      <td>0.480940</td>\n",
       "      <td>1.368103</td>\n",
       "      <td>0.452177</td>\n",
       "      <td>0.180113</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Gene Symbol  Abundances (Normalized): F1: Sample, Control  \\\n",
       "0         VIM                                  1.649179e+10   \n",
       "1       AHNAK                                  2.003021e+09   \n",
       "2        ACTB                                  5.672231e+09   \n",
       "3       ACTG1                                  2.238417e+06   \n",
       "4       GAPDH                                  1.242975e+09   \n",
       "\n",
       "   Abundances (Normalized): F2: Sample, Control  \\\n",
       "0                                  1.661164e+10   \n",
       "1                                  1.849071e+09   \n",
       "2                                  5.636443e+09   \n",
       "3                                  2.190261e+06   \n",
       "4                                  1.282440e+09   \n",
       "\n",
       "   Abundances (Normalized): F3: Sample, Control  \\\n",
       "0                                  1.524631e+10   \n",
       "1                                  1.713771e+09   \n",
       "2                                  5.878443e+09   \n",
       "3                                  2.738121e+06   \n",
       "4                                  1.185116e+09   \n",
       "\n",
       "   Abundances (Normalized): F4: Sample, Etoposide  \\\n",
       "0                                    1.194626e+10   \n",
       "1                                    1.772634e+09   \n",
       "2                                    7.444967e+09   \n",
       "3                                    1.120958e+06   \n",
       "4                                    1.199184e+09   \n",
       "\n",
       "   Abundances (Normalized): F5: Sample, Etoposide  \\\n",
       "0                                    8.138632e+09   \n",
       "1                                    2.325231e+09   \n",
       "2                                    5.882621e+09   \n",
       "3                                    1.139723e+06   \n",
       "4                                    8.301472e+08   \n",
       "\n",
       "   Abundances (Normalized): F6: Sample, Etoposide  \\\n",
       "0                                    8.359638e+09   \n",
       "1                                    2.399826e+09   \n",
       "2                                    5.694489e+09   \n",
       "3                                    1.601739e+06   \n",
       "4                                    9.464704e+08   \n",
       "\n",
       "   Abundances (Normalized): F10: Sample, FTY720  \\\n",
       "0                                  1.037863e+10   \n",
       "1                                  2.122741e+09   \n",
       "2                                  5.942383e+09   \n",
       "3                                  3.598291e+06   \n",
       "4                                  8.816873e+08   \n",
       "\n",
       "   Abundances (Normalized): F11: Sample, FTY720  \\\n",
       "0                                  1.260664e+10   \n",
       "1                                  2.138561e+09   \n",
       "2                                  7.232767e+09   \n",
       "3                                  1.953038e+06   \n",
       "4                                  9.276548e+08   \n",
       "\n",
       "   Abundances (Normalized): F12: Sample, FTY720  ...  log2_FC_FTY/CTRL  \\\n",
       "0                                  1.303859e+10  ...         -0.430605   \n",
       "1                                  2.067453e+09  ...          0.188090   \n",
       "2                                  6.858702e+09  ...          0.216463   \n",
       "3                                  6.898486e+05  ...         -0.489795   \n",
       "4                                  1.222988e+09  ...         -0.305769   \n",
       "\n",
       "   FC_FTY_Eto/CTRL  log2_FC_FTY_Eto/CTRL  FC_FTY/Eto  log2_FC_FTY/Eto  \\\n",
       "0         0.805703             -0.311680    1.280363         0.356552   \n",
       "1         1.075813              0.105428    0.982645        -0.025258   \n",
       "2         1.087792              0.121403    1.057319         0.080411   \n",
       "3         0.713667             -0.486676    1.333092         0.414776   \n",
       "4         1.106810              0.146408    1.020137         0.028763   \n",
       "\n",
       "   FC_FTY_Eto/Eto  log2_FC_FTY_Eto/Eto  FC_FTY_Eto/FTY  log2_FC_FTY_Eto/FTY  \\\n",
       "0        1.390379             0.475478        1.085926             0.118926   \n",
       "1        0.927924            -0.107921        0.944313            -0.082663   \n",
       "2        0.989898            -0.014649        0.936233            -0.095060   \n",
       "3        1.335976             0.417895        1.002164             0.003118   \n",
       "4        1.395653             0.480940        1.368103             0.452177   \n",
       "\n",
       "      ANOVA  \n",
       "0  0.005436  \n",
       "1  0.249711  \n",
       "2  0.294790  \n",
       "3  0.347499  \n",
       "4  0.180113  \n",
       "\n",
       "[5 rows x 31 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteome.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cfde9ec5",
   "metadata": {},
   "source": [
    "# Start from here for Tukey"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "05dd5d23",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'proteome' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/px/yhcbpgrs1hv_2v25cgzl51w00000gn/T/ipykernel_1691/1923248881.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0msig_proteome\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mproteome\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mproteome\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"ANOVA\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m<\u001b[0m \u001b[0;36m0.05\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msig_proteome\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0msig_proteome\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_excel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"/Users/mortezaabyadeh/Desktop/proteome_Anova_Significant.xlsx\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'proteome' is not defined"
     ]
    }
   ],
   "source": [
    "sig_proteome = proteome[proteome[\"ANOVA\"]< 0.05]\n",
    "print(sig_proteome.shape)\n",
    "\n",
    "sig_proteome.to_excel(\"/Users/mortezaabyadeh/Desktop/proteome_Anova_Significant.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "4a660055",
   "metadata": {},
   "outputs": [],
   "source": [
    "sig_proteome = pd.read_excel(\"/Users/mortezaabyadeh/Desktop/proteome_Anova_Significant.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e4be66b0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>Abundances (Normalized): F1: Sample, Control</th>\n",
       "      <th>Abundances (Normalized): F2: Sample, Control</th>\n",
       "      <th>Abundances (Normalized): F3: Sample, Control</th>\n",
       "      <th>Abundances (Normalized): F4: Sample, Etoposide</th>\n",
       "      <th>Abundances (Normalized): F5: Sample, Etoposide</th>\n",
       "      <th>Abundances (Normalized): F6: Sample, Etoposide</th>\n",
       "      <th>Abundances (Normalized): F10: Sample, FTY720</th>\n",
       "      <th>Abundances (Normalized): F11: Sample, FTY720</th>\n",
       "      <th>...</th>\n",
       "      <th>log2_FC_FTY/CTRL</th>\n",
       "      <th>FC_FTY_Eto/CTRL</th>\n",
       "      <th>log2_FC_FTY_Eto/CTRL</th>\n",
       "      <th>FC_FTY/Eto</th>\n",
       "      <th>log2_FC_FTY/Eto</th>\n",
       "      <th>FC_FTY_Eto/Eto</th>\n",
       "      <th>log2_FC_FTY_Eto/Eto</th>\n",
       "      <th>FC_FTY_Eto/FTY</th>\n",
       "      <th>log2_FC_FTY_Eto/FTY</th>\n",
       "      <th>ANOVA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>VIM</td>\n",
       "      <td>1.649179e+10</td>\n",
       "      <td>1.661164e+10</td>\n",
       "      <td>1.524631e+10</td>\n",
       "      <td>1.194626e+10</td>\n",
       "      <td>8.138632e+09</td>\n",
       "      <td>8.359638e+09</td>\n",
       "      <td>1.037863e+10</td>\n",
       "      <td>1.260664e+10</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.430605</td>\n",
       "      <td>0.805703</td>\n",
       "      <td>-0.311680</td>\n",
       "      <td>1.280363</td>\n",
       "      <td>0.356552</td>\n",
       "      <td>1.390379</td>\n",
       "      <td>0.475478</td>\n",
       "      <td>1.085926</td>\n",
       "      <td>0.118926</td>\n",
       "      <td>0.005436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6</td>\n",
       "      <td>FLNA</td>\n",
       "      <td>1.884852e+09</td>\n",
       "      <td>1.839619e+09</td>\n",
       "      <td>1.935036e+09</td>\n",
       "      <td>2.638653e+09</td>\n",
       "      <td>2.835520e+09</td>\n",
       "      <td>2.820679e+09</td>\n",
       "      <td>2.111028e+09</td>\n",
       "      <td>2.516056e+09</td>\n",
       "      <td>...</td>\n",
       "      <td>0.308271</td>\n",
       "      <td>1.256850</td>\n",
       "      <td>0.329813</td>\n",
       "      <td>0.845099</td>\n",
       "      <td>-0.242808</td>\n",
       "      <td>0.857812</td>\n",
       "      <td>-0.221267</td>\n",
       "      <td>1.015044</td>\n",
       "      <td>0.021542</td>\n",
       "      <td>0.000744</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>1.569636e+09</td>\n",
       "      <td>1.631728e+09</td>\n",
       "      <td>1.595765e+09</td>\n",
       "      <td>1.673058e+09</td>\n",
       "      <td>1.884370e+09</td>\n",
       "      <td>1.881159e+09</td>\n",
       "      <td>1.649051e+09</td>\n",
       "      <td>1.627710e+09</td>\n",
       "      <td>...</td>\n",
       "      <td>0.036440</td>\n",
       "      <td>1.031495</td>\n",
       "      <td>0.044737</td>\n",
       "      <td>0.905886</td>\n",
       "      <td>-0.142598</td>\n",
       "      <td>0.911111</td>\n",
       "      <td>-0.134301</td>\n",
       "      <td>1.005768</td>\n",
       "      <td>0.008297</td>\n",
       "      <td>0.024251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8</td>\n",
       "      <td>FLNC</td>\n",
       "      <td>8.821027e+08</td>\n",
       "      <td>9.208451e+08</td>\n",
       "      <td>8.932544e+08</td>\n",
       "      <td>7.766220e+08</td>\n",
       "      <td>7.912183e+08</td>\n",
       "      <td>7.933552e+08</td>\n",
       "      <td>7.903533e+08</td>\n",
       "      <td>7.063110e+08</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.266628</td>\n",
       "      <td>0.894686</td>\n",
       "      <td>-0.160546</td>\n",
       "      <td>0.949087</td>\n",
       "      <td>-0.075388</td>\n",
       "      <td>1.021504</td>\n",
       "      <td>0.030694</td>\n",
       "      <td>1.076302</td>\n",
       "      <td>0.106082</td>\n",
       "      <td>0.001289</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>EEF1A1</td>\n",
       "      <td>1.992194e+09</td>\n",
       "      <td>1.924985e+09</td>\n",
       "      <td>2.081250e+09</td>\n",
       "      <td>1.744530e+09</td>\n",
       "      <td>1.844523e+09</td>\n",
       "      <td>1.736457e+09</td>\n",
       "      <td>1.726034e+09</td>\n",
       "      <td>1.109267e+09</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.654124</td>\n",
       "      <td>0.750452</td>\n",
       "      <td>-0.414169</td>\n",
       "      <td>0.715662</td>\n",
       "      <td>-0.482650</td>\n",
       "      <td>0.845166</td>\n",
       "      <td>-0.242694</td>\n",
       "      <td>1.180956</td>\n",
       "      <td>0.239955</td>\n",
       "      <td>0.021051</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0 Gene Symbol  Abundances (Normalized): F1: Sample, Control  \\\n",
       "0           0         VIM                                  1.649179e+10   \n",
       "1           6        FLNA                                  1.884852e+09   \n",
       "2           7        MYH9                                  1.569636e+09   \n",
       "3           8        FLNC                                  8.821027e+08   \n",
       "4           9      EEF1A1                                  1.992194e+09   \n",
       "\n",
       "   Abundances (Normalized): F2: Sample, Control  \\\n",
       "0                                  1.661164e+10   \n",
       "1                                  1.839619e+09   \n",
       "2                                  1.631728e+09   \n",
       "3                                  9.208451e+08   \n",
       "4                                  1.924985e+09   \n",
       "\n",
       "   Abundances (Normalized): F3: Sample, Control  \\\n",
       "0                                  1.524631e+10   \n",
       "1                                  1.935036e+09   \n",
       "2                                  1.595765e+09   \n",
       "3                                  8.932544e+08   \n",
       "4                                  2.081250e+09   \n",
       "\n",
       "   Abundances (Normalized): F4: Sample, Etoposide  \\\n",
       "0                                    1.194626e+10   \n",
       "1                                    2.638653e+09   \n",
       "2                                    1.673058e+09   \n",
       "3                                    7.766220e+08   \n",
       "4                                    1.744530e+09   \n",
       "\n",
       "   Abundances (Normalized): F5: Sample, Etoposide  \\\n",
       "0                                    8.138632e+09   \n",
       "1                                    2.835520e+09   \n",
       "2                                    1.884370e+09   \n",
       "3                                    7.912183e+08   \n",
       "4                                    1.844523e+09   \n",
       "\n",
       "   Abundances (Normalized): F6: Sample, Etoposide  \\\n",
       "0                                    8.359638e+09   \n",
       "1                                    2.820679e+09   \n",
       "2                                    1.881159e+09   \n",
       "3                                    7.933552e+08   \n",
       "4                                    1.736457e+09   \n",
       "\n",
       "   Abundances (Normalized): F10: Sample, FTY720  \\\n",
       "0                                  1.037863e+10   \n",
       "1                                  2.111028e+09   \n",
       "2                                  1.649051e+09   \n",
       "3                                  7.903533e+08   \n",
       "4                                  1.726034e+09   \n",
       "\n",
       "   Abundances (Normalized): F11: Sample, FTY720  ...  log2_FC_FTY/CTRL  \\\n",
       "0                                  1.260664e+10  ...         -0.430605   \n",
       "1                                  2.516056e+09  ...          0.308271   \n",
       "2                                  1.627710e+09  ...          0.036440   \n",
       "3                                  7.063110e+08  ...         -0.266628   \n",
       "4                                  1.109267e+09  ...         -0.654124   \n",
       "\n",
       "   FC_FTY_Eto/CTRL  log2_FC_FTY_Eto/CTRL  FC_FTY/Eto  log2_FC_FTY/Eto  \\\n",
       "0         0.805703             -0.311680    1.280363         0.356552   \n",
       "1         1.256850              0.329813    0.845099        -0.242808   \n",
       "2         1.031495              0.044737    0.905886        -0.142598   \n",
       "3         0.894686             -0.160546    0.949087        -0.075388   \n",
       "4         0.750452             -0.414169    0.715662        -0.482650   \n",
       "\n",
       "   FC_FTY_Eto/Eto  log2_FC_FTY_Eto/Eto  FC_FTY_Eto/FTY  log2_FC_FTY_Eto/FTY  \\\n",
       "0        1.390379             0.475478        1.085926             0.118926   \n",
       "1        0.857812            -0.221267        1.015044             0.021542   \n",
       "2        0.911111            -0.134301        1.005768             0.008297   \n",
       "3        1.021504             0.030694        1.076302             0.106082   \n",
       "4        0.845166            -0.242694        1.180956             0.239955   \n",
       "\n",
       "      ANOVA  \n",
       "0  0.005436  \n",
       "1  0.000744  \n",
       "2  0.024251  \n",
       "3  0.001289  \n",
       "4  0.021051  \n",
       "\n",
       "[5 rows x 32 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig_proteome.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "40ce51ac",
   "metadata": {},
   "source": [
    "# With all information\n",
    "\n",
    "\n",
    "control_cols = [\"Abundances (Normalized): F1: Sample, Control\", \"Abundances (Normalized): F2: Sample, Control\", \"Abundances (Normalized): F3: Sample, Control\"]\n",
    "etoposide_cols = [\"Abundances (Normalized): F4: Sample, Etoposide\", \"Abundances (Normalized): F5: Sample, Etoposide\", \"Abundances (Normalized): F6: Sample, Etoposide\"]\n",
    "fty_cols = [\"Abundances (Normalized): F10: Sample, FTY720\", \"Abundances (Normalized): F11: Sample, FTY720\", \"Abundances (Normalized): F12: Sample, FTY720\"]\n",
    "fty_eto_cols = [\"Abundances (Normalized): F7: Sample, FTY720_plus_Etoposide\", \"Abundances (Normalized): F8: Sample, FTY720_plus_Etoposide\",\"Abundances (Normalized): F9: Sample, FTY720_plus_Etoposide\"]\n",
    "\n",
    "\n",
    "all_cols = control_cols + etoposide_cols + fty_cols + fty_eto_cols\n",
    "\n",
    "# Subset the data\n",
    "expression_data = sig_proteome[all_cols]\n",
    "genes = sig_proteome[\"Gene Symbol\"]\n",
    "\n",
    "\n",
    "# Tukey's post-hoc test and save each comparison in separate sheets\n",
    "with pd.ExcelWriter('/Users/mortezaabyadeh/Desktop/posthoc_results.xlsx') as writer:\n",
    "    for comparison_pair in [('Control', 'Etoposide'), ('Control', 'FTY720'), ('Control', 'FTY720_plus_Etoposide'),\n",
    "                            ('Etoposide', 'FTY720'), ('Etoposide', 'FTY720_plus_Etoposide'), ('FTY720', 'FTY720_plus_Etoposide')]:\n",
    "        comparison_results = []\n",
    "        for i in range(len(expression_data)):\n",
    "            data_i = expression_data.iloc[i, :].values\n",
    "            labels = np.repeat([\"Control\", \"Etoposide\", \"FTY720\", \"FTY720_plus_Etoposide\"], 3)\n",
    "            tukey = pairwise_tukeyhsd(data_i, labels)\n",
    "            results_table = pd.DataFrame(data=tukey.summary().data[1:], columns=tukey.summary().data[0])\n",
    "            results_table['Gene'] = genes.iloc[i]\n",
    "            comparison_results.append(results_table[results_table['group1'].isin(comparison_pair) & results_table['group2'].isin(comparison_pair)])\n",
    "        \n",
    "        comparison_df = pd.concat(comparison_results)\n",
    "        comparison_df.to_excel(writer, sheet_name=f'{comparison_pair[0]}_vs_{comparison_pair[1]}', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "8059ba04",
   "metadata": {},
   "outputs": [],
   "source": [
    "# I have filtered only adj-pvalue\n",
    "control_cols = [\"Abundances (Normalized): F1: Sample, Control\", \"Abundances (Normalized): F2: Sample, Control\", \"Abundances (Normalized): F3: Sample, Control\"]\n",
    "etoposide_cols = [\"Abundances (Normalized): F4: Sample, Etoposide\", \"Abundances (Normalized): F5: Sample, Etoposide\", \"Abundances (Normalized): F6: Sample, Etoposide\"]\n",
    "fty_cols = [\"Abundances (Normalized): F10: Sample, FTY720\", \"Abundances (Normalized): F11: Sample, FTY720\", \"Abundances (Normalized): F12: Sample, FTY720\"]\n",
    "fty_eto_cols = [\"Abundances (Normalized): F7: Sample, FTY720_plus_Etoposide\", \"Abundances (Normalized): F8: Sample, FTY720_plus_Etoposide\", \"Abundances (Normalized): F9: Sample, FTY720_plus_Etoposide\"]\n",
    "\n",
    "all_cols = control_cols + etoposide_cols + fty_cols + fty_eto_cols\n",
    "\n",
    "expression_data = sig_proteome[all_cols]\n",
    "genes = sig_proteome[\"Gene Symbol\"]\n",
    "\n",
    "Tukey's post-hoc test and save each comparison in separate sheets\n",
    "with pd.ExcelWriter('/Users/mortezaabyadeh/Desktop/posthoc_results.xlsx') as writer:\n",
    "    for comparison_pair in [('Control', 'Etoposide'), ('Control', 'FTY720'), ('Control', 'FTY720_Eto'),\n",
    "                            ('Etoposide', 'FTY720'), ('Etoposide', 'FTY720_Eto'), ('FTY720', 'FTY720_Eto')]:\n",
    "        comparison_results = []\n",
    "        for i in range(len(expression_data)):\n",
    "            data_i = expression_data.iloc[i, :].values\n",
    "            labels = np.repeat([\"Control\", \"Etoposide\", \"FTY720\", \"FTY720_Eto\"], 3)\n",
    "            tukey = pairwise_tukeyhsd(data_i, labels)\n",
    "            results_table = pd.DataFrame(data=tukey.summary().data[1:], columns=tukey.summary().data[0])\n",
    "            results_table['Gene'] = genes.iloc[i]\n",
    "            comparison_results.append(results_table[results_table['group1'].isin(comparison_pair) & results_table['group2'].isin(comparison_pair)][['Gene', 'p-adj']])\n",
    "        \n",
    "        comparison_df = pd.concat(comparison_results)\n",
    "        comparison_df.to_excel(writer, sheet_name=f'{comparison_pair[0]}_vs_{comparison_pair[1]}', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a951bed6",
   "metadata": {},
   "source": [
    "# Finally worked after the the whole day running and getting issues! <h1>Congratulations! &#x1F389;</h1>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "ada740e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "sig_proteome.columns\n",
    "posthoc_results2 = pd.read_excel(\"/Users/mortezaabyadeh/Desktop/posthoc_results 2.xlsx\", sheet_name=None)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "340f64d0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception ignored in: <function ZipFile.__del__ at 0x7fc8f05dc160>\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/anaconda3/lib/python3.9/zipfile.py\", line 1816, in __del__\n",
      "    self.close()\n",
      "  File \"/opt/anaconda3/lib/python3.9/zipfile.py\", line 1833, in close\n",
      "    self.fp.seek(self.start_dir)\n",
      "ValueError: seek of closed file\n"
     ]
    }
   ],
   "source": [
    "start_column = 19\n",
    "end_column = 30\n",
    "step = 2\n",
    "\n",
    "with pd.ExcelWriter('/Users/mortezaabyadeh/Desktop/posthoc_results2.xlsx', engine='openpyxl') as writer:\n",
    "    for sheet_name, df in posthoc_results.items():\n",
    "        # Initialize index for accessing columns in sig_proteome\n",
    "        col_index = start_column\n",
    "        while col_index <= end_column:\n",
    "            # Get the column pair from sig_proteome\n",
    "            col1 = sig_proteome.columns[col_index]\n",
    "            col2 = sig_proteome.columns[col_index + 1]\n",
    "            \n",
    "            # Add the columns from sig_proteome to the current sheet\n",
    "            df[f'Additional1_{col1}_{col2}'] = sig_proteome[col1]\n",
    "            df[f'Additional2_{col1}_{col2}'] = sig_proteome[col2]\n",
    "            \n",
    "            # Move to the next pair of columns\n",
    "            col_index += step\n",
    "        \n",
    "        df.to_excel(writer, sheet_name=sheet_name, index=False)\n",
    "\n",
    "print(\"Additional columns added and saved successfully.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "86467470",
   "metadata": {},
   "outputs": [],
   "source": [
    "start_column = 19\n",
    "end_column = 30\n",
    "step = 2\n",
    "\n",
    "with pd.ExcelWriter('/Users/mortezaabyadeh/Desktop/posthoc_results2.xlsx', engine='openpyxl') as writer:\n",
    "    for sheet_name, df in posthoc_results2.items():\n",
    "        # Initialize index for accessing columns in sig_proteome\n",
    "        col_index = start_column\n",
    "        while col_index <= end_column:\n",
    "            # Get the column pair from sig_proteome\n",
    "            col1 = sig_proteome.columns[col_index]\n",
    "            col2 = sig_proteome.columns[col_index + 1]\n",
    "            \n",
    "            # Add the columns from sig_proteome to the current sheet\n",
    "            df[f'{col1}'] = sig_proteome[col1]\n",
    "            df[f'{col2}'] = sig_proteome[col2]\n",
    "            \n",
    "            # Move to the next pair of columns\n",
    "            col_index += step\n",
    "        \n",
    "        df.to_excel(writer, sheet_name=sheet_name, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "c1d975f0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtered data saved to /Users/mortezaabyadeh/Desktop/posthoc_results_final_filtered.xlsx\n"
     ]
    }
   ],
   "source": [
    "file_path = '/Users/mortezaabyadeh/Desktop/posthoc_results_final.xlsx'\n",
    "xl = pd.ExcelFile(file_path)\n",
    "\n",
    "# Dictionary to store filtered DataFrames for each sheet\n",
    "filtered_dfs = {}\n",
    "\n",
    "# Iterate through each sheet\n",
    "for sheet_name in xl.sheet_names:\n",
    "    df = xl.parse(sheet_name)\n",
    "    \n",
    "    df_filtered = df[df['p-adj'] < 0.05]\n",
    "    \n",
    "    filtered_dfs[sheet_name] = df_filtered\n",
    "\n",
    "output_path = '/Users/mortezaabyadeh/Desktop/posthoc_results_final_filtered.xlsx'\n",
    "with pd.ExcelWriter(\"/Users/mortezaabyadeh/Desktop/posthoc_results_final.xlsx\", engine='openpyxl') as writer:\n",
    "    for sheet_name, df_filtered in filtered_dfs.items():\n",
    "        df_filtered.to_excel(writer, sheet_name=sheet_name, index=False)\n",
    "\n",
    "print(f\"Filtered data saved to {output_path}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "43ac4e30",
   "metadata": {},
   "source": [
    "# Start from here using the posthoc_sig file, first upload it!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "bcfd1d9d",
   "metadata": {},
   "outputs": [],
   "source": [
    "proteome_data = pd.read_excel(\"/Users/mortezaabyadeh/Desktop/posthoc_results_final.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "35ddf2e0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>p-adj</th>\n",
       "      <th>FC_Eto/CTRL</th>\n",
       "      <th>log2_FC_Eto/CTRL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>VIM</td>\n",
       "      <td>0.0014</td>\n",
       "      <td>0.579485</td>\n",
       "      <td>-0.787158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FLNA</td>\n",
       "      <td>0.0001</td>\n",
       "      <td>1.465182</td>\n",
       "      <td>0.551080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MYH9</td>\n",
       "      <td>0.0210</td>\n",
       "      <td>1.132129</td>\n",
       "      <td>0.179038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FLNC</td>\n",
       "      <td>0.0051</td>\n",
       "      <td>0.875852</td>\n",
       "      <td>-0.191240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACTA2</td>\n",
       "      <td>0.0008</td>\n",
       "      <td>5.196467</td>\n",
       "      <td>2.377531</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene   p-adj  FC_Eto/CTRL  log2_FC_Eto/CTRL\n",
       "0    VIM  0.0014     0.579485         -0.787158\n",
       "1   FLNA  0.0001     1.465182          0.551080\n",
       "2   MYH9  0.0210     1.132129          0.179038\n",
       "3   FLNC  0.0051     0.875852         -0.191240\n",
       "4  ACTA2  0.0008     5.196467          2.377531"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteome_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "43ae7bfc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Separated data saved to /Users/mortezaabyadeh/Desktop/posthoc_results_separated.xlsx\n"
     ]
    }
   ],
   "source": [
    "file_path = '/Users/mortezaabyadeh/Desktop/posthoc_results_final.xlsx'\n",
    "xl = pd.ExcelFile(file_path)\n",
    "\n",
    "# Create a new Excel writer object to save the separated data\n",
    "output_path = '/Users/mortezaabyadeh/Desktop/posthoc_results_separated.xlsx'\n",
    "with pd.ExcelWriter(output_path, engine='openpyxl') as writer:\n",
    "    # Iterate through each sheet\n",
    "    for sheet_name in xl.sheet_names:\n",
    "        # Read the sheet into a DataFrame\n",
    "        df = xl.parse(sheet_name)\n",
    "        \n",
    "        # Identify the column with 'log2_FC_' prefix\n",
    "        log2_fc_col = [col for col in df.columns if col.startswith('log2_FC')][0]\n",
    "        \n",
    "        # Separate rows based on positive or negative values in the log2_FC column\n",
    "        df_negative = df[df[log2_fc_col] < 0]\n",
    "        df_positive = df[df[log2_fc_col] >= 0]\n",
    "        \n",
    "        df_negative.to_excel(writer, sheet_name=f'{sheet_name}_negative', index=False)\n",
    "        df_positive.to_excel(writer, sheet_name=f'{sheet_name}_positive', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ebc7e825",
   "metadata": {},
   "source": [
    "# Trancriptome data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "23c7d8f4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: biopython in /opt/anaconda3/lib/python3.9/site-packages (1.83)\n",
      "Requirement already satisfied: numpy in /opt/anaconda3/lib/python3.9/site-packages (from biopython) (1.24.4)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install biopython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "498ecc91",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "702b7cca",
   "metadata": {},
   "outputs": [],
   "source": [
    "# records = list(SeqIO.parse(\"/Users/mortezaabyadeh/Desktop/HS_1_1.fq\", \"fastq\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d61f77c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_df = pd.read_excel(\"/Users/mortezaabyadeh/Desktop/gene_count.xlsx\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ba09f6b8",
   "metadata": {},
   "source": [
    "# def calculate_fpkm(counts_df):\n",
    "    gene_lengths = counts_df['gene_length']  \n",
    "    counts = counts_df.drop(columns=['gene_name', 'gene_length'])\n",
    "    \n",
    "    # Calculate the total counts per sample\n",
    "    total_counts = counts.sum(axis=0)\n",
    "    \n",
    "    # Calculate FPKM\n",
    "    fpkm = (counts.T / (gene_lengths * 1e-3)).T\n",
    "    fpkm = fpkm.div(total_counts * 1e-6, axis=1)\n",
    "    \n",
    "    return fpkm\n",
    "\n",
    "fpkm_df = calculate_fpkm(counts_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "44cf4a26",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pip in /opt/anaconda3/lib/python3.9/site-packages (22.2.2)\n",
      "Collecting pip\n",
      "  Downloading pip-24.1.2-py3-none-any.whl (1.8 MB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.8/1.8 MB\u001b[0m \u001b[31m15.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: setuptools in /opt/anaconda3/lib/python3.9/site-packages (63.4.1)\n",
      "Collecting setuptools\n",
      "  Using cached setuptools-70.2.0-py3-none-any.whl (930 kB)\n",
      "Requirement already satisfied: wheel in /opt/anaconda3/lib/python3.9/site-packages (0.37.1)\n",
      "Collecting wheel\n",
      "  Using cached wheel-0.43.0-py3-none-any.whl (65 kB)\n",
      "Installing collected packages: wheel, setuptools, pip\n",
      "  Attempting uninstall: wheel\n",
      "    Found existing installation: wheel 0.37.1\n",
      "    Uninstalling wheel-0.37.1:\n",
      "      Successfully uninstalled wheel-0.37.1\n",
      "  Attempting uninstall: setuptools\n",
      "    Found existing installation: setuptools 63.4.1\n",
      "    Uninstalling setuptools-63.4.1:\n",
      "      Successfully uninstalled setuptools-63.4.1\n",
      "  Attempting uninstall: pip\n",
      "    Found existing installation: pip 22.2.2\n",
      "    Uninstalling pip-22.2.2:\n",
      "      Successfully uninstalled pip-22.2.2\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "spyder 5.3.3 requires pyqt5<5.16, which is not installed.\n",
      "spyder 5.3.3 requires pyqtwebengine<5.16, which is not installed.\n",
      "anaconda-project 0.11.1 requires ruamel-yaml, which is not installed.\n",
      "conda-repo-cli 1.0.20 requires clyent==1.2.1, but you have clyent 1.2.2 which is incompatible.\n",
      "conda-repo-cli 1.0.20 requires nbformat==5.4.0, but you have nbformat 5.5.0 which is incompatible.\n",
      "conda-repo-cli 1.0.20 requires requests==2.28.1, but you have requests 2.31.0 which is incompatible.\u001b[0m\u001b[31m\n",
      "\u001b[0mSuccessfully installed pip-24.1.2 setuptools-70.2.0 wheel-0.43.0\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install --upgrade pip setuptools wheel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2a9e9928",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting rpy2\n",
      "  Downloading rpy2-3.5.16-cp39-cp39-macosx_11_0_x86_64.whl.metadata (4.5 kB)\n",
      "Collecting cffi>=1.15.1 (from rpy2)\n",
      "  Downloading cffi-1.16.0-cp39-cp39-macosx_10_9_x86_64.whl.metadata (1.5 kB)\n",
      "Collecting jinja2 (from rpy2)\n",
      "  Using cached jinja2-3.1.4-py3-none-any.whl.metadata (2.6 kB)\n",
      "Collecting tzlocal (from rpy2)\n",
      "  Using cached tzlocal-5.2-py3-none-any.whl.metadata (7.8 kB)\n",
      "Collecting pycparser (from cffi>=1.15.1->rpy2)\n",
      "  Using cached pycparser-2.22-py3-none-any.whl.metadata (943 bytes)\n",
      "Collecting MarkupSafe>=2.0 (from jinja2->rpy2)\n",
      "  Downloading MarkupSafe-2.1.5-cp39-cp39-macosx_10_9_x86_64.whl.metadata (3.0 kB)\n",
      "Downloading rpy2-3.5.16-cp39-cp39-macosx_11_0_x86_64.whl (260 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m260.8/260.8 kB\u001b[0m \u001b[31m8.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hDownloading cffi-1.16.0-cp39-cp39-macosx_10_9_x86_64.whl (182 kB)\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m182.5/182.5 kB\u001b[0m \u001b[31m11.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hUsing cached jinja2-3.1.4-py3-none-any.whl (133 kB)\n",
      "Using cached tzlocal-5.2-py3-none-any.whl (17 kB)\n",
      "Downloading MarkupSafe-2.1.5-cp39-cp39-macosx_10_9_x86_64.whl (14 kB)\n",
      "Using cached pycparser-2.22-py3-none-any.whl (117 kB)\n",
      "Installing collected packages: tzlocal, pycparser, MarkupSafe, jinja2, cffi, rpy2\n",
      "  Attempting uninstall: pycparser\n",
      "    Found existing installation: pycparser 2.21\n",
      "    Uninstalling pycparser-2.21:\n",
      "      Successfully uninstalled pycparser-2.21\n",
      "  Attempting uninstall: MarkupSafe\n",
      "    Found existing installation: MarkupSafe 2.1.5\n",
      "    Uninstalling MarkupSafe-2.1.5:\n",
      "      Successfully uninstalled MarkupSafe-2.1.5\n",
      "  Attempting uninstall: jinja2\n",
      "    Found existing installation: Jinja2 3.1.4\n",
      "    Uninstalling Jinja2-3.1.4:\n",
      "      Successfully uninstalled Jinja2-3.1.4\n",
      "  Attempting uninstall: cffi\n",
      "    Found existing installation: cffi 1.15.1\n",
      "    Uninstalling cffi-1.15.1:\n",
      "      Successfully uninstalled cffi-1.15.1\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "anaconda-project 0.11.1 requires ruamel-yaml, which is not installed.\n",
      "spyder 5.3.3 requires pyqt5<5.16, which is not installed.\n",
      "spyder 5.3.3 requires pyqtwebengine<5.16, which is not installed.\n",
      "d3graph 2.5.0 requires markupsafe==2.0.1, but you have markupsafe 2.1.5 which is incompatible.\n",
      "distributed 2022.7.0 requires dask==2022.7.0, but you have dask 2024.6.2 which is incompatible.\n",
      "pandas-profiling 3.1.0 requires joblib~=1.0.1, but you have joblib 1.3.2 which is incompatible.\n",
      "pandas-profiling 3.1.0 requires markupsafe~=2.0.1, but you have markupsafe 2.1.5 which is incompatible.\n",
      "ydata-profiling 4.8.3 requires visions[type_image_path]<0.7.7,>=0.7.5, but you have visions 0.7.4 which is incompatible.\u001b[0m\u001b[31m\n",
      "\u001b[0mSuccessfully installed MarkupSafe-2.1.5 cffi-1.16.0 jinja2-3.1.4 pycparser-2.22 rpy2-3.5.16 tzlocal-5.2\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install --force-reinstall rpy2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "599a32c2",
   "metadata": {},
   "outputs": [],
   "source": [
    "import rpy2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9c6b8c4c",
   "metadata": {},
   "source": [
    "# Not working, I went through R and used DESeq2 package!\n",
    "\n",
    "import pandas as pd\n",
    "from rpy2.robjects import pandas2ri, r\n",
    "from rpy2.robjects.packages import importr\n",
    "\n",
    "# Convert pandas DataFrame to R dataframe\n",
    "pandas2ri.activate()\n",
    "r_dataframe = pandas2ri.py2ri(counts_df)\n",
    "\n",
    "# Load DESeq2 from R\n",
    "deseq2 = importr('DESeq2')\n",
    "\n",
    "# Design formula\n",
    "design_formula = \"~ group\"\n",
    "\n",
    "# DESeq2 analysis\n",
    "deseq_dataset = deseq2.DESeqDataSetFromMatrix(countData=r_dataframe,\n",
    "                                              colData=col_data,\n",
    "                                              design=design_formula)\n",
    "\n",
    "# Run DESeq2 analysis\n",
    "deseq_dataset = deseq2.DESeq(deseq_dataset)\n",
    "\n",
    "# Get results\n",
    "results = deseq2.results(deseq_dataset)\n",
    "\n",
    "# Extract differential expression results\n",
    "deseq_results = deseq2.results(results)\n",
    "\n",
    "# Convert results back to pandas DataFrame\n",
    "deseq_results_df = pandas2ri.ri2py_dataframe(deseq_results)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "91addde8",
   "metadata": {},
   "outputs": [],
   "source": [
    "if (!requireNamespace(\"BiocManager\", quietly = TRUE))\n",
    "  install.packages(\"BiocManager\")\n",
    "BiocManager::install(\"DESeq2\")\n",
    "library(DESeq2)\n",
    "library(readxl)\n",
    "library(openxlsx)\n",
    "\n",
    "counts_df <- read_excel(\"/Users/mortezaabyadeh/Desktop/gene_count.xlsx\")\n",
    "head(counts_df)\n",
    "\n",
    "sample_names <- colnames(counts_df)[3:ncol(counts_df)] # Extract sample names from counts_df\n",
    "\n",
    "# Create col_data with sample names and group information\n",
    "col_data <- data.frame(\n",
    "  sampleName = sample_names,\n",
    "  group = factor(rep(c(\"CTRL\", \"Eto\", \"Dox\", \"FTY\", \"FTY_Dox\", \"FTY_Eto\", \"Eto_FTY\", \"Dox_FTY\"), each = 3))\n",
    ")\n",
    "\n",
    "countData <- counts_df[, -c(1, 2)]\n",
    "\n",
    "# Create DESeqDataSet object\n",
    "dds <- DESeqDataSetFromMatrix(countData = countData,\n",
    "                              colData = col_data,\n",
    "                              design = ~ group)\n",
    "dds <- DESeq(dds)\n",
    "\n",
    "\n",
    "results <- results(dds)\n",
    "\n",
    "\n",
    "results_Eto_vs_CTRL <- results(dds, contrast = c(\"group\", \"Eto\", \"CTRL\"))\n",
    "results_FTY_vs_CTRL <- results(dds, contrast = c(\"group\", \"FTY\", \"CTRL\"))\n",
    "results_FTY_Eto_vs_CTRL <- results(dds, contrast = c(\"group\", \"FTY_Eto\", \"CTRL\"))\n",
    "results_FTY_vs_Eto <- results(dds, contrast = c(\"group\", \"FTY\", \"Eto\"))\n",
    "results_FTY_Eto_vs_Eto <- results(dds, contrast = c(\"group\", \"FTY_Eto\", \"Eto\"))\n",
    "results_FTY_Eto_vs_FTY <- results(dds, contrast = c(\"group\", \"FTY_Eto\", \"FTY\"))\n",
    "\n",
    "gene_names <- counts_df$gene_name\n",
    "\n",
    "\n",
    "results_Eto_vs_CTRL <- cbind(gene_name = gene_names, results_Eto_vs_CTRL)\n",
    "results_FTY_vs_CTRL <- cbind(gene_name = gene_names, results_FTY_vs_CTRL)\n",
    "results_FTY_Eto_vs_CTRL <- cbind(gene_name = gene_names, results_FTY_Eto_vs_CTRL)\n",
    "results_FTY_vs_Eto <- cbind(gene_name = gene_names, results_FTY_vs_Eto)\n",
    "results_FTY_Eto_vs_Eto <- cbind(gene_name = gene_names, results_FTY_Eto_vs_Eto)\n",
    "results_FTY_Eto_vs_FTY <- cbind(gene_name = gene_names, results_FTY_Eto_vs_FTY)\n",
    "\n",
    "significant_eto_ctrl <- subset(results_Eto_vs_CTRL, padj < 0.05)\n",
    "significant_fty_ctrl <- subset(results_FTY_vs_CTRL, padj < 0.05)\n",
    "significant_fty_eto_ctrl <- subset(results_FTY_Eto_vs_CTRL, padj < 0.05)\n",
    "significant_fty_eto <- subset(results_FTY_vs_Eto, padj < 0.05)\n",
    "significant_fty_eto_eto <- subset(results_FTY_Eto_vs_Eto, padj < 0.05)\n",
    "significant_fty_eto_fty <- subset(results_FTY_Eto_vs_FTY, padj < 0.05)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "library(openxlsx)\n",
    "\n",
    "excel_file <- \"/Users/mortezaabyadeh/Desktop/DEGs.xlsx\"\n",
    "\n",
    "wb <- createWorkbook()\n",
    "\n",
    "addWorksheet(wb, sheetName = \"Eto_vs_CTRL\")\n",
    "writeData(wb, sheet = \"Eto_vs_CTRL\", significant_eto_ctrl)\n",
    "\n",
    "addWorksheet(wb, sheetName = \"FTY_vs_CTRL\")\n",
    "writeData(wb, sheet = \"FTY_vs_CTRL\", significant_fty_ctrl)\n",
    "\n",
    "addWorksheet(wb, sheetName = \"FTY_Eto_vs_CTRL\")\n",
    "writeData(wb, sheet = \"FTY_Eto_vs_CTRL\", significant_fty_eto_ctrl)\n",
    "\n",
    "addWorksheet(wb, sheetName = \"FTY_vs_Eto\")\n",
    "writeData(wb, sheet = \"FTY_vs_Eto\", significant_fty_eto)\n",
    "\n",
    "addWorksheet(wb, sheetName = \"FTY_Eto_vs_Eto\")\n",
    "writeData(wb, sheet = \"FTY_Eto_vs_Eto\", significant_fty_eto_eto)\n",
    "\n",
    "addWorksheet(wb, sheetName = \"FTY_Eto_vs_FTY\")\n",
    "writeData(wb, sheet = \"FTY_Eto_vs_FTY\", significant_fty_eto_fty)\n",
    "\n",
    "saveWorkbook(wb, excel_file)\n",
    "\n",
    "\n",
    "\n",
    "filterAndWrite <- function(data, sheet_name) {\n",
    "  Down <- subset(data, log2FoldChange < 0)\n",
    "  \n",
    "  Up <- subset(data, log2FoldChange > 0)\n",
    "  \n",
    "  addWorksheet(wb, sheetName = paste0(sheet_name, \"_Down\"))\n",
    "  writeData(wb, sheet = paste0(sheet_name, \"_Down\"), Down)\n",
    "  \n",
    "  addWorksheet(wb, sheetName = paste0(sheet_name, \"_Up\"))\n",
    "  writeData(wb, sheet = paste0(sheet_name, \"_Up\"), Up)\n",
    "}\n",
    "\n",
    "filterAndWrite(significant_eto_ctrl, \"Eto_vs_CTRL\")\n",
    "filterAndWrite(significant_fty_ctrl, \"FTY_vs_CTRL\")\n",
    "filterAndWrite(significant_fty_eto_ctrl, \"FTY_Eto_vs_CTRL\")\n",
    "filterAndWrite(significant_fty_eto, \"FTY_vs_Eto\")\n",
    "filterAndWrite(significant_fty_eto_eto, \"FTY_Eto_vs_Eto\")\n",
    "filterAndWrite(significant_fty_eto_fty, \"FTY_Eto_vs_FTY\")\n",
    "\n",
    "saveWorkbook(wb, \"/Users/mortezaabyadeh/Desktop/DEGs_final.xlsx\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
